# Functional Class Scoring {#functional-class-scoring-1}

```{r include=FALSE}
library(knitr)
opts_chunk$set(message = FALSE, warning = FALSE, eval = TRUE, echo = TRUE, cache = TRUE)
library(genekitr)
```

[ORA method](#ora-go-analysis) is easy to conduct, but it will lose information when genes' differences are slight. For example, if a gene is essential in some pathway but is low-expressed, it will be filtered by fold change cutoff.

Unlike ORA, FCS tools do not set a threshold for differentially expressed genes. Instead, it gives each detected gene a differential expression score and then evaluates whether the scores are more positive or negative than expected by chance for each gene set.

> According to article "[Ten Years of Pathway Analysis: Current Approaches and Outstanding Challenges](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3285573/)":

FCS approaches include [GSEA](https://www.gsea-msigdb.org/), [GlobalTest](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2712740/), [sigPathway](https://pubmed.ncbi.nlm.nih.gov/16174746), [SAFE](https://pubmed.ncbi.nlm.nih.gov/15647293), [GSA](https://scholar.google.com/scholar_lookup?journal=Ann+Appl+Stat&title=On+testing+the+significance+of+sets+of+genes&author=B+Efron&author=R+Tibshirani&volume=1&publication_year=2007&pages=107-129&), [PADOG](https://pubmed.ncbi.nlm.nih.gov/22713124), [PCOT2](https://pubmed.ncbi.nlm.nih.gov/16877751), [FunCluster](https://pubmed.ncbi.nlm.nih.gov/17007070), [SAM-GS](https://pubmed.ncbi.nlm.nih.gov/17612399), [Category](https://pubmed.ncbi.nlm.nih.gov/17127676) etc.

Next, we will go through the popular Gene Set Enrichment Analysis (GSEA) tool, which uses FCS permutation approaches to determine whether a gene set is significantly associated with higher or lower scores.

## GSEA intruduction {#gsea-introduction}

> To do GSEA, user must have an ordered gene rank list (e.g., genes with decreasing logFC order).

Three steps in GSEA:

1. Calculate the enrichment score (ES): represents the amount to which the genes in the set are over-represented at either the top or bottom of the list. GSEA starts at the top of the ranked gene list to calculate the enrichment score. If a gene is a member of the candidate gene set, it adds to a running sum. Otherwise, it subtracts.

2. Estimate the statistical significance of the ES: this calculation is done by a phenotypic-based permutation test in order to produce a null distribution for the ES.

3. Adjust for multiple hypothesis testing for when a large number of gene sets are being analyzed at one time: the enrichment scores for each set are normalized, and a false discovery rate is calculated.

(ref:gseaoverviewScap) GSEA overview

(ref:gseaoverviewCap) **GSEA overview**

```{r gseaoverview, out.width="100%", echo=FALSE, fig.cap="(ref:gseaoverviewCap)", fig.scap="(ref:gseaoverviewScap)"}
knitr::include_graphics("figures/gsea_overview.png")
```

> If you are interested in GSEA detailed procedure, please visit: https://www.pathwaycommons.org/guide/primers/data_analysis/gsea

[Msigdb categories](http://www.gsea-msigdb.org/gsea/msigdb/collections.jsp) is the best GSEA partner which have 9 major collections and several sub-collections from 32880 gene sets: 

+ H: hallmark gene sets (50 gene sets)
+ C1: positional gene sets (299 gene sets)
  - by chromosome: chr1 => MT
+ C2: curated gene sets (6366 gene sets)
  - CGP (chemical and genetic perturbations, 3384 gene sets)
  - CP (canonical pathways, 2982 gene sets) includes BioCarta, KEGG, PID, Reactome and WikiPathways
+ C3: regulatory target gene sets (3726 gene sets)
  - MIR (microRNA targets, 2598 gene sets)
  - TFT (all transcription factor targets, 1128 gene sets)
+ C4: computational gene sets (858 gene sets)
  - CGN (cancer gene neighborhoods, 427 gene sets)
  - CM (cancer modules, 431 gene sets)
+ C5: ontology gene sets (15473 gene sets) includes BP, CC and MF
+ C6: oncogenic signature gene sets (189 gene sets)
+ C7: immunologic signature gene sets (5219 gene sets)
  - IMMUNESIGDB (ImmuneSigDB gene sets, 4872 gene sets)
  - VAX (vaccine response gene sets, 347 gene sets)
+ C8: cell type signature gene sets (700 gene sets)


## Basic usage {#gsea-basic-usage}

> `genKEGG` is built on `msigdbr` and `clusterProfiler::GSEA`, then optimizes analysis process to make better understanding. 

The simplest arguments are: 

- `genelist`: Order ranked genelist in decreasing order, gene can be entrez, ensembl or symbol.
- `org`: organism (from `genekitr::msig_org`)
- `category`: one of C1','C2','C3', 'C4','C5','C6','C7','C8','H'
- `subcategory`: choose from `genekitr::msig_category`. If category has no subcategory, leave it as NULL.
- `pvalueCutoff`
- `qvalueCutoff`

```{r gsea-basic-usage}
data(geneList, package = "genekitr")
gse <- genGSEA(
  genelist = geneList,
  org = "human",
  category = "H",
  use_symbol = TRUE
)
class(gse)
names(gse)
```

The GSEA result is a list including input genelist, target gene set and analysis result.

```{r gsea-check-result}
head(gse$genelist)
head(gse$geneset)
head(gse$gsea_df, 5)
```




