<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 10 Functional Class Scoring | Genekitr: Gene Analysis Toolkit based on R</title>
<meta name="author" content="Yunze Liu">
<meta name="description" content="ORA method is easy to conduct, but it will lose information when genes‚Äô differences are slight. For example, if a gene is essential in some pathway but is low-expressed, it will be filtered by...">
<meta name="generator" content="bookdown 0.26 with bs4_book()">
<meta property="og:title" content="Chapter 10 Functional Class Scoring | Genekitr: Gene Analysis Toolkit based on R">
<meta property="og:type" content="book">
<meta property="og:description" content="ORA method is easy to conduct, but it will lose information when genes‚Äô differences are slight. For example, if a gene is essential in some pathway but is low-expressed, it will be filtered by...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 10 Functional Class Scoring | Genekitr: Gene Analysis Toolkit based on R">
<meta name="twitter:description" content="ORA method is easy to conduct, but it will lose information when genes‚Äô differences are slight. For example, if a gene is essential in some pathway but is low-expressed, it will be filtered by...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.22/datatables.js"></script><link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Genekitr: Gene Analysis Toolkit based on R</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">üß¨ Welcome to genekitr!</a></li>
<li><a class="" href="quick-start.html"><span class="header-section-number">1</span> Quick start</a></li>
<li class="book-part">Part I: Search Information</li>
<li><a class="" href="search-gene-metadata-1.html"><span class="header-section-number">2</span> Search gene metadata</a></li>
<li><a class="" href="search-pubmed-1.html"><span class="header-section-number">3</span> Search PubMed records</a></li>
<li class="book-part">Part II: ID Conversion</li>
<li><a class="" href="gene-id-conversion-1.html"><span class="header-section-number">4</span> Gene ID conversion</a></li>
<li><a class="" href="protein-id-conversion-1.html"><span class="header-section-number">5</span> Protein ID conversion</a></li>
<li><a class="" href="probe-id-conversion-1.html"><span class="header-section-number">6</span> Probe ID conversion</a></li>
<li class="book-part">Part III: Enrichment Analysis</li>
<li><a class="" href="ora-go-analysis-1.html"><span class="header-section-number">7</span> ORA GO analysis</a></li>
<li><a class="" href="ora-kegg-analysis-1.html"><span class="header-section-number">8</span> ORA KEGG analysis</a></li>
<li><a class="" href="ora-additional-1.html"><span class="header-section-number">9</span> ORA Additional</a></li>
<li><a class="active" href="functional-class-scoring-1.html"><span class="header-section-number">10</span> Functional Class Scoring</a></li>
<li><a class="" href="import-external-data-1.html"><span class="header-section-number">11</span> Import External Data</a></li>
<li class="book-part">Part IV: Plotting</li>
<li><a class="" href="plot-go-kegg-1.html"><span class="header-section-number">12</span> Plot GO and KEGG</a></li>
<li><a class="" href="plot-gsea-1.html"><span class="header-section-number">13</span> Plot GSEA</a></li>
<li><a class="" href="plot-others-1.html"><span class="header-section-number">14</span> Plot others</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="functional-class-scoring-1" class="section level1">
<h1>
<span class="header-section-number">10</span> Functional Class Scoring<a class="anchor" aria-label="anchor" href="#functional-class-scoring-1"><i class="fas fa-link"></i></a>
</h1>
<p><a href="quick-start.html#ora-go-analysis">ORA method</a> is easy to conduct, but it will lose information when genes‚Äô differences are slight. For example, if a gene is essential in some pathway but is low-expressed, it will be filtered by fold change cutoff.</p>
<p>Unlike ORA, FCS tools do not set a threshold for differentially expressed genes. Instead, it gives each detected gene a differential expression score and then evaluates whether the scores are more positive or negative than expected by chance for each gene set.</p>
<blockquote>
<p>According to article ‚Äú<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3285573/">Ten Years of Pathway Analysis: Current Approaches and Outstanding Challenges</a>‚Äù:</p>
</blockquote>
<p>FCS approaches include <a href="https://www.gsea-msigdb.org/">GSEA</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2712740/">GlobalTest</a>, <a href="https://pubmed.ncbi.nlm.nih.gov/16174746">sigPathway</a>, <a href="https://pubmed.ncbi.nlm.nih.gov/15647293">SAFE</a>, <a href="https://scholar.google.com/scholar_lookup?journal=Ann+Appl+Stat&amp;title=On+testing+the+significance+of+sets+of+genes&amp;author=B+Efron&amp;author=R+Tibshirani&amp;volume=1&amp;publication_year=2007&amp;pages=107-129&amp;">GSA</a>, <a href="https://pubmed.ncbi.nlm.nih.gov/22713124">PADOG</a>, <a href="https://pubmed.ncbi.nlm.nih.gov/16877751">PCOT2</a>, <a href="https://pubmed.ncbi.nlm.nih.gov/17007070">FunCluster</a>, <a href="https://pubmed.ncbi.nlm.nih.gov/17612399">SAM-GS</a>, <a href="https://pubmed.ncbi.nlm.nih.gov/17127676">Category</a> etc.</p>
<p>Next, we will go through the popular Gene Set Enrichment Analysis (GSEA) tool, which uses FCS permutation approaches to determine whether a gene set is significantly associated with higher or lower scores.</p>
<div id="gsea-introduction" class="section level2">
<h2>
<span class="header-section-number">10.1</span> GSEA intruduction<a class="anchor" aria-label="anchor" href="#gsea-introduction"><i class="fas fa-link"></i></a>
</h2>
<blockquote>
<p>To do GSEA, user must have an ordered gene rank list (e.g., genes with decreasing logFC order).</p>
</blockquote>
<p>Three steps in GSEA:</p>
<ol style="list-style-type: decimal">
<li><p>Calculate the enrichment score (ES): represents the amount to which the genes in the set are over-represented at either the top or bottom of the list. GSEA starts at the top of the ranked gene list to calculate the enrichment score. If a gene is a member of the candidate gene set, it adds to a running sum. Otherwise, it subtracts.</p></li>
<li><p>Estimate the statistical significance of the ES: this calculation is done by a phenotypic-based permutation test in order to produce a null distribution for the ES.</p></li>
<li><p>Adjust for multiple hypothesis testing for when a large number of gene sets are being analyzed at one time: the enrichment scores for each set are normalized, and a false discovery rate is calculated.</p></li>
</ol>
<div class="figure">
<span style="display:block;" id="fig:gseaoverview"></span>
<img src="figures/gsea_overview.png" alt="GSEA overview" width="100%"><p class="caption">
Figure 10.1: <strong>GSEA overview</strong>
</p>
</div>
<blockquote>
<p>If you are interested in GSEA detailed procedure, please visit: <a href="https://www.pathwaycommons.org/guide/primers/data_analysis/gsea" class="uri">https://www.pathwaycommons.org/guide/primers/data_analysis/gsea</a> OR <a href="https://github.com/crazyhottommy/RNA-seq-analysis/blob/master/GSEA_explained.md" class="uri">https://github.com/crazyhottommy/RNA-seq-analysis/blob/master/GSEA_explained.md</a></p>
</blockquote>
<p><a href="http://www.gsea-msigdb.org/gsea/msigdb/collections.jsp">Msigdb categories</a> is the best GSEA partner which have 9 major collections and several sub-collections from 32880 gene sets:</p>
<ul>
<li>H: hallmark gene sets (50 gene sets)</li>
<li>C1: positional gene sets (299 gene sets)
<ul>
<li>by chromosome: chr1 =&gt; MT</li>
</ul>
</li>
<li>C2: curated gene sets (6366 gene sets)
<ul>
<li>CGP (chemical and genetic perturbations, 3384 gene sets)</li>
<li>CP (canonical pathways, 2982 gene sets) includes BioCarta, KEGG, PID, Reactome and WikiPathways</li>
</ul>
</li>
<li>C3: regulatory target gene sets (3726 gene sets)
<ul>
<li>MIR (microRNA targets, 2598 gene sets)</li>
<li>TFT (all transcription factor targets, 1128 gene sets)</li>
</ul>
</li>
<li>C4: computational gene sets (858 gene sets)
<ul>
<li>CGN (cancer gene neighborhoods, 427 gene sets)</li>
<li>CM (cancer modules, 431 gene sets)</li>
</ul>
</li>
<li>C5: ontology gene sets (15473 gene sets) includes BP, CC and MF</li>
<li>C6: oncogenic signature gene sets (189 gene sets)</li>
<li>C7: immunologic signature gene sets (5219 gene sets)
<ul>
<li>IMMUNESIGDB (ImmuneSigDB gene sets, 4872 gene sets)</li>
<li>VAX (vaccine response gene sets, 347 gene sets)</li>
</ul>
</li>
<li>C8: cell type signature gene sets (700 gene sets)</li>
</ul>
</div>
<div id="gsea-basic-usage" class="section level2">
<h2>
<span class="header-section-number">10.2</span> Basic usage<a class="anchor" aria-label="anchor" href="#gsea-basic-usage"><i class="fas fa-link"></i></a>
</h2>
<blockquote>
<p><code>genKEGG</code> is built on <code>msigdbr</code> and <code><a href="https://rdrr.io/pkg/clusterProfiler/man/GSEA.html">clusterProfiler::GSEA</a></code> which is also built on <code>fgsea</code>. Also <code>genKEGG</code> optimizes analysis process to make better performance.</p>
</blockquote>
<p>The simplest arguments are:</p>
<ul>
<li>
<code>genelist</code>: Order ranked genelist in decreasing order, gene can be <strong>Entrez, Ensembl or symbol.</strong>
</li>
<li>
<code>org</code>: organism (from <code><a href="https://rdrr.io/pkg/genekitr/man/Datasets.html">genekitr::msig_org</a></code>)</li>
<li>
<code>category</code>: one of C1‚Äô,‚ÄòC2‚Äô,‚ÄòC3‚Äô, ‚ÄòC4‚Äô,‚ÄòC5‚Äô,‚ÄòC6‚Äô,‚ÄòC7‚Äô,‚ÄòC8‚Äô,‚ÄòH‚Äô</li>
<li>
<code>subcategory</code>: choose from <code><a href="https://rdrr.io/pkg/genekitr/man/Datasets.html">genekitr::msig_category</a></code>. If category has no subcategory, leave it as NULL.</li>
<li><code>pvalueCutoff</code></li>
<li><code>qvalueCutoff</code></li>
</ul>
<div class="sourceCode" id="cb199"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneList</span>, package <span class="op">=</span> <span class="st">"genekitr"</span><span class="op">)</span>
<span class="va">gse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/genGSEA.html">genGSEA</a></span><span class="op">(</span>
  genelist <span class="op">=</span> <span class="va">geneList</span>,
  org <span class="op">=</span> <span class="st">"human"</span>,
  category <span class="op">=</span> <span class="st">"H"</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "list"</code></pre>
<div class="sourceCode" id="cb201"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "gsea_df"  "genelist" "geneset"  "exponent" "org"</code></pre>
<p>The GSEA result is a list including input genelist, target gene set and analysis result.</p>
<div class="sourceCode" id="cb203"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gse</span><span class="op">$</span><span class="va">genelist</span><span class="op">)</span></code></pre></div>
<pre><code>##                  ID    logfc
## 948             948 5.780170
## 1638           1638 5.633027
## 158471       158471 4.683610
## 10610         10610 3.875120
## 6947           6947 3.357670
## 100133941 100133941 3.322533</code></pre>
<div class="sourceCode" id="cb205"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gse</span><span class="op">$</span><span class="va">geneset</span><span class="op">)</span></code></pre></div>
<pre><code>##                 gs_name entrez_gene
## 1 HALLMARK_ADIPOGENESIS          19
## 2 HALLMARK_ADIPOGENESIS       11194
## 3 HALLMARK_ADIPOGENESIS       10449
## 4 HALLMARK_ADIPOGENESIS          33
## 5 HALLMARK_ADIPOGENESIS          34
## 6 HALLMARK_ADIPOGENESIS          35</code></pre>
<div class="sourceCode" id="cb207"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gse</span><span class="op">$</span><span class="va">gsea_df</span>, <span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code>##                                                ID              Description
## HALLMARK_E2F_TARGETS         HALLMARK_E2F_TARGETS     HALLMARK_E2F_TARGETS
## HALLMARK_G2M_CHECKPOINT   HALLMARK_G2M_CHECKPOINT  HALLMARK_G2M_CHECKPOINT
## HALLMARK_MYC_TARGETS_V1   HALLMARK_MYC_TARGETS_V1  HALLMARK_MYC_TARGETS_V1
## HALLMARK_MYC_TARGETS_V2   HALLMARK_MYC_TARGETS_V2  HALLMARK_MYC_TARGETS_V2
## HALLMARK_MITOTIC_SPINDLE HALLMARK_MITOTIC_SPINDLE HALLMARK_MITOTIC_SPINDLE
##                          setSize enrichmentScore       NES       pvalue
## HALLMARK_E2F_TARGETS         194      -0.8361902 -3.447027 1.000000e-10
## HALLMARK_G2M_CHECKPOINT      195      -0.8028510 -3.306427 1.000000e-10
## HALLMARK_MYC_TARGETS_V1      187      -0.7348931 -3.028913 1.000000e-10
## HALLMARK_MYC_TARGETS_V2       54      -0.8644875 -2.956731 1.000000e-10
## HALLMARK_MITOTIC_SPINDLE     197      -0.5161722 -2.132762 1.029192e-10
##                              p.adjust       qvalue rank
## HALLMARK_E2F_TARGETS     1.029192e-09 4.333439e-10 2116
## HALLMARK_G2M_CHECKPOINT  1.029192e-09 4.333439e-10 2114
## HALLMARK_MYC_TARGETS_V1  1.029192e-09 4.333439e-10 3480
## HALLMARK_MYC_TARGETS_V2  1.029192e-09 4.333439e-10 2138
## HALLMARK_MITOTIC_SPINDLE 1.029192e-09 4.333439e-10 2192
##                                            leading_edge
## HALLMARK_E2F_TARGETS     tags=79%, list=11%, signal=71%
## HALLMARK_G2M_CHECKPOINT  tags=72%, list=11%, signal=64%
## HALLMARK_MYC_TARGETS_V1  tags=70%, list=19%, signal=57%
## HALLMARK_MYC_TARGETS_V2  tags=87%, list=12%, signal=77%
## HALLMARK_MITOTIC_SPINDLE tags=32%, list=12%, signal=29%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             geneID
## HALLMARK_E2F_TARGETS     9126/5885/6117/3609/5511/9183/204/23047/2935/3184/6426/4292/5981/1854/7514/11200/11051/6749/6427/6434/9125/11113/3159/8243/5036/1033/10492/7283/5411/1965/9738/1786/6628/10274/7037/3930/9238/5889/898/7913/1434/9133/3838/9972/79077/7083/10212/991/51155/10051/3978/5901/9232/4678/8914/9787/5558/84844/79019/3148/64785/7884/11340/5810/5424/24137/2146/9221/81611/7374/10606/1111/10535/580/7153/55148/5631/57122/3835/1633/1164/3161/29127/4436/5347/10248/6839/10460/1062/332/23165/10797/6790/5426/6118/7398/29893/27338/10615/7465/11004/64858/55635/9700/55143/83461/5902/81624/5982/9212/4288/23649/10714/4609/29089/1163/5111/146909/147841/4605/4001/7112/1503/84296/672/4172/54962/9833/4171/4176/5983/6941/79075/79733/4174/4085/9837/4173/983/4175/55723/9319/3070/57405/10733/253714/55646/23594/29028/675/10635/6241/993
## HALLMARK_G2M_CHECKPOINT                                                                                    5885/4502/6421/3192/3609/996/7290/9994/3312/9252/23047/2935/3184/51776/6426/3619/10951/7514/4928/6427/6434/7027/3159/8243/1033/9184/10492/11044/10432/3980/4953/8317/10419/22823/10772/7468/7150/10274/4088/3930/81539/4691/1871/4957/9133/3838/6713/10146/3837/10212/991/51155/10051/1063/10592/387893/9232/4678/5558/51203/6541/1841/24137/2176/2146/9221/10762/1111/10024/580/7153/10926/4751/3835/6632/10403/1164/22974/3161/29127/262/5347/6839/10460/57082/1062/1870/332/6790/5426/1058/6118/10293/27338/9055/3832/11004/11065/9700/5933/899/9212/4288/9585/699/23649/890/4609/1946/1736/1163/4605/4001/7112/990/4172/9493/7272/10036/7371/55872/1869/26271/56992/4171/6491/10721/4174/4085/983/4175/10733/23594/8438/675/51659/595/8318/993/9156
## HALLMARK_MYC_TARGETS_V1                                                                                                                                        2091/5230/5093/23450/26121/65005/6742/8454/1933/10935/5682/1982/5713/7419/5478/2547/5887/1176/5690/10155/10574/6637/3068/6741/5707/3608/10907/10549/5425/5688/6626/26156/56910/7416/10236/4686/5704/3192/11137/3939/9868/11335/1977/6432/5709/10921/2935/3184/23016/5706/6426/3066/1854/7514/5245/7307/8894/6427/2107/6434/708/57819/7027/7411/5036/6428/4706/8669/9184/10492/3329/10291/7555/6950/790/1965/4953/10575/10213/10054/10856/7203/6634/2739/10728/26135/7965/7913/3838/1964/10146/10576/3837/5496/991/8662/5901/6732/1478/6629/3251/9221/54107/8886/1665/22948/6240/6632/9136/26354/7398/6059/9188/1017/5902/5634/5984/51491/890/4609/7298/5111/1503/4171/4176/4174/4085/4173/4175/8318
## HALLMARK_MYC_TARGETS_V2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       6832/10244/11335/10196/79050/5245/79711/5036/3329/10199/81887/9238/2193/7965/54663/27346/79077/23481/64216/51388/10171/4839/27340/56915/80324/23160/9221/8886/92856/7374/83743/5347/6573/9136/29078/10514/26354/23082/705/6949/3099/51491/4609/51154/4174/4173/10733
## HALLMARK_MITOTIC_SPINDLE                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              3799/5116/10160/9126/9793/55722/996/10300/79658/4643/3619/55201/55755/396/11064/58526/5981/5829/8243/5922/357/10426/10611/382/9133/55835/10051/1063/9732/6624/9787/51203/10435/24137/10276/7153/50628/4751/3835/10403/22974/29127/5347/1062/332/6790/54443/9055/9735/3832/11004/1894/9700/9585/699/163786/4001/9493/7272/26271/56992/983/675
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            geneID_symbol
## HALLMARK_E2F_TARGETS     SMC3/RAD21/RPA1/ILF3/PPP1R8/ZW10/AK2/PDS5B/GSPT1/HNRNPD/SRSF1/MLH1/RFC1/DUT/XPO1/CHEK2/NUDT21/SSRP1/SRSF2/TRA2B/CNOT9/CIT/HMGA1/SMC1A/PA2G4/CDKN3/SYNCRIP/TUBG1/PNN/EIF2S1/CCP110/DNMT1/SNRPB/STAG1/TFRC/LBR/TBRG4/RAD51C/CCNE1/DEK/CSE1L/CCNB2/KPNA2/NUP153/DCTPP1/TK1/DDX39A/CDC20/JPT1/SMC4/LIG1/RAN/PTTG1/NASP/TIMELESS/DLGAP5/PRIM2/PHF5A/CENPM/HMGB2/GINS3/SLBP/EXOSC8/RAD1/POLD1/KIF4A/EZH2/NOLC1/ANP32E/UNG/PAICS/CHEK1/RNASEH2A/BARD1/TOP2A/UBR7/PRPS1/NUP107/KIF22/DCK/CKS2/HMMR/RACGAP1/MSH2/PLK1/POP7/SUV39H1/TACC3/CENPE/BIRC5/NUP205/MTHFD2/AURKA/POLE/RPA2/USP1/PSMC3IP/UBE2S/SPAG5/WEE1/KIF2C/DCLRE1B/DEPDC1/ESPL1/CDCA8/CDCA3/RANBP1/DIAPH3/RFC2/AURKB/MKI67/POLA2/POLD3/MYC/UBE2T/CKS1B/PCNA/KIF18B/SPC24/MYBL2/LMNB1/TMPO/CTPS1/GINS4/BRCA1/MCM3/TIPIN/MELK/MCM2/MCM7/RFC3/TCF19/DSCC1/E2F8/MCM5/MAD2L1/GINS1/MCM4/CDK1/MCM6/ASF1B/TRIP13/HELLS/SPC25/PLK4/MMS22L/LYAR/ORC6/ATAD2/BRCA2/RAD51AP1/RRM2/CDC25A
## HALLMARK_G2M_CHECKPOINT                                                                                          RAD21/MT2A/SFPQ/HNRNPU/ILF3/CDC27/HIRA/CASP8AP2/HSPA8/RPS6KA5/PDS5B/GSPT1/HNRNPD/MAP3K20/SRSF1/INCENP/CBX1/XPO1/NUP98/SRSF2/TRA2B/TFDP1/HMGA1/SMC1A/CDKN3/BUB3/SYNCRIP/TENT4A/RBM14/LIG3/ODC1/CDC7/PRMT5/MTF2/SRSF10/NSD2/TOP1/STAG1/SMAD3/LBR/SLC38A1/NCL/E2F3/ODF2/CCNB2/KPNA2/SQLE/G3BP1/KPNB1/DDX39A/CDC20/JPT1/SMC4/CENPF/SMC2/KMT5A/PTTG1/NASP/PRIM2/NUSAP1/SLC7A1/DTYMK/KIF4A/FANCC/EZH2/NOLC1/NUP50/CHEK1/TROAP/BARD1/TOP2A/DBF4/NEK2/KIF22/SNRPD1/NDC80/CKS2/TPX2/HMMR/RACGAP1/AMD1/PLK1/SUV39H1/TACC3/KNL1/CENPE/E2F2/BIRC5/AURKA/POLE/CENPA/RPA2/TRAIP/UBE2S/PRC1/KIF11/KIF2C/UBE2C/ESPL1/RBL1/CCNF/AURKB/MKI67/KIF20B/BUB1/POLA2/CCNA2/MYC/EFNA5/DKC1/CKS1B/MYBL2/LMNB1/TMPO/CDC6/MCM3/KIF23/TTK/CHAF1A/UCK2/PBK/E2F1/FBXO5/KIF15/MCM2/STIL/POLQ/MCM5/MAD2L1/CDK1/MCM6/PLK4/ORC6/RAD54L/BRCA2/GINS2/CCND1/CDC45/CDC25A/EXO1
## HALLMARK_MYC_TARGETS_V1                                                                                                                                             FBL/PGK1/PCBP1/SF3B3/PRPF31/MRPL9/SSBP1/CUL1/EEF1B2/PRDX3/PSMA1/EIF4G2/PSMD7/VDAC3/PPIA/XRCC6/RAD23B/AP3S1/PSMB2/TRIM28/CCT7/SNRPG/HDGF/SSB/PSMD1/ILF2/TXNL4A/PRDX4/POLD2/PSMA7/SNRPA/RSL1D1/STARD7/VDAC1/HNRNPR/NCBP1/PSMC4/HNRNPU/PWP1/LDHA/TOMM70/CBX3/EIF4E/SRSF7/PSMD3/RNPS1/GSPT1/HNRNPD/EXOSC7/PSMC6/SRSF1/HDAC2/DUT/XPO1/PHB/U2AF1/EIF2S2/SRSF2/ETF1/TRA2B/C1QBP/LSM2/TFDP1/VBP1/PA2G4/SRSF3/NDUFAB1/EIF3J/BUB3/SYNCRIP/HSPD1/SF3A1/CNBP/TCP1/CAD/EIF2S1/ODC1/CCT4/PSMD14/UBA2/RUVBL2/CCT3/SNRPD3/GLO1/PTGES3/SERBP1/AIMP2/DEK/KPNA2/EIF1AX/G3BP1/CCT2/KPNB1/PPM1G/CDC20/EIF3B/RAN/SRPK1/CSTF2/SNRPB2/HPRT1/NOLC1/POLE3/DDX18/DHX15/CCT5/RRM1/SNRPD1/RRP9/GNL3/USP1/ABCE1/DDX21/CDK2/RANBP1/PRPS2/RFC4/NOP16/CCNA2/MYC/TYMS/PCNA/CTPS1/MCM2/MCM7/MCM5/MAD2L1/MCM4/MCM6/CDC45
## HALLMARK_MYC_TARGETS_V2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               SUPV3L1/RABEPK/CBX3/PRMT3/NOC4L/PHB/IPO4/PA2G4/HSPD1/MPHOSPH10/LAS1L/TBRG4/FARSA/AIMP2/WDR74/TMEM97/DCTPP1/PES1/TFB2M/NIP7/RCL1/NOP2/UTP20/EXOSC5/PUS1/WDR43/NOLC1/DDX18/IMP4/UNG/GRWD1/PLK1/SLC19A1/RRP9/NDUFAF4/MYBBP1A/GNL3/PPRC1/BYSL/TCOF1/HK2/NOP16/MYC/MRTO4/MCM5/MCM4/PLK4
## HALLMARK_MITOTIC_SPINDLE                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               KIF5B/PCNT/FARP1/SMC3/CKAP5/CEP72/CDC27/KATNB1/ARHGAP10/MYO1E/INCENP/MAP1S/CDK5RAP2/ARHGDIA/CNTRL/MID1IP1/RFC1/PXN/SMC1A/RASA2/SHROOM2/TUBGCP3/PDLIM5/ARF6/CCNB2/CENPJ/SMC4/CENPF/DOCK4/FSCN1/DLGAP5/NUSAP1/CDC42EP2/KIF4A/NET1/TOP2A/GEMIN4/NEK2/KIF22/NDC80/TPX2/RACGAP1/PLK1/CENPE/BIRC5/AURKA/ANLN/PRC1/KNTC1/KIF11/KIF2C/ECT2/ESPL1/KIF20B/BUB1/SASS6/LMNB1/KIF23/TTK/FBXO5/KIF15/CDK1/BRCA2
##                          Count
## HALLMARK_E2F_TARGETS       153
## HALLMARK_G2M_CHECKPOINT    140
## HALLMARK_MYC_TARGETS_V1    130
## HALLMARK_MYC_TARGETS_V2     47
## HALLMARK_MITOTIC_SPINDLE    63</code></pre>
<blockquote>
<p>About the <code>gsea_df</code> result</p>
</blockquote>
<ul>
<li><p>If input id is mixture of gene symbol and alias, the result will convert alias to symbol automatically and result will show two columns <code>geneID</code> and <code>geneID_symbol</code>. <strong>If input id is symbol and the result only has <code>geneID</code>, that means the input symbol do not have any alias.</strong></p></li>
<li><p><code>setSize</code>: according to the <a href="https://rdrr.io/github/ctlab/fgsea/man/fgseaLabel.html">documentation</a> of <a href="http://bioconductor.org/packages/fgsea">fgsea</a>, it is the size of the pathway after removing genes not present in <code>names(stats)</code>. To be more specific, if we input a pathway gene set with 58 genes (e.g.¬†<a href="https://www.gsea-msigdb.org/gsea/msigdb/cards/HALLMARK_MYC_TARGETS_V2.html">HALLMARK_MYC_TARGETS_V2</a> ), while our gene list only have 54 of it, so the result will only show <code>setSize = 54</code></p></li>
<li><p><code>enrichmentScore</code>: also called <code>ES</code>, same as in <a href="https://www.gsea-msigdb.org/gsea/doc/GSEAUserGuideFrame.html?Interpreting_GSEA">Broad GSEA implementation</a>. It reflects the degree to which a gene set is overrepresented at the top or bottom of a ranked list of genes.</p></li>
<li><p><code>NES</code>: normalized enrichment score is the primary statistic for examining gene set enrichment results. By normalizing the enrichment score, GSEA accounts for differences in gene set size and in correlations between gene sets and the expression dataset. Therefore, NES can be used to compare analysis results across gene sets.</p></li>
</ul>
<p>A positive normalized enrichment scores (NES) will indicate that genes in set S will be mostly represented at the top of your list (logFC &gt; 0 or up-regulated genes).</p>
</div>
<div id="export-result" class="section level2">
<h2>
<span class="header-section-number">10.3</span> Export result<a class="anchor" aria-label="anchor" href="#export-result"><i class="fas fa-link"></i></a>
</h2>
<p><code>Genekitr</code> sets an easy way to export analysis result for editing and sharing. User could save multiple data sets as different sheets in one excel file. Besides, colmun names are automatically formated with bold style.</p>
<div class="sourceCode" id="cb209"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">genekitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/expoSheet.html">expoSheet</a></span><span class="op">(</span>data_list <span class="op">=</span> <span class="va">gse</span>,
                    data_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span>,
                    filename <span class="op">=</span> <span class="st">"gsea_result.xlsx"</span>,
                    dir <span class="op">=</span> <span class="st">"./"</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span style="display:block;" id="fig:exposheet"></span>
<img src="figures/exposheet.png" alt="Export result" width="100%"><p class="caption">
Figure 10.2: <strong>Export result</strong>
</p>
</div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="ora-additional-1.html"><span class="header-section-number">9</span> ORA Additional</a></div>
<div class="next"><a href="import-external-data-1.html"><span class="header-section-number">11</span> Import External Data</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#functional-class-scoring-1"><span class="header-section-number">10</span> Functional Class Scoring</a></li>
<li><a class="nav-link" href="#gsea-introduction"><span class="header-section-number">10.1</span> GSEA intruduction</a></li>
<li><a class="nav-link" href="#gsea-basic-usage"><span class="header-section-number">10.2</span> Basic usage</a></li>
<li><a class="nav-link" href="#export-result"><span class="header-section-number">10.3</span> Export result</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Genekitr: Gene Analysis Toolkit based on R</strong>" was written by Yunze Liu. It was last built on 2022-06-27.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
