<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 4 Gene ID conversion | Genekitr: Gene Analysis Toolkit based on R</title>
<meta name="author" content="Yunze Liu">
<meta name="description" content="If you have handled genomics analysis, then you may spend much time converting gene identifiers to other types. For example, gene set enrichment analysis is very common in omics downstream process...">
<meta name="generator" content="bookdown 0.26 with bs4_book()">
<meta property="og:title" content="Chapter 4 Gene ID conversion | Genekitr: Gene Analysis Toolkit based on R">
<meta property="og:type" content="book">
<meta property="og:description" content="If you have handled genomics analysis, then you may spend much time converting gene identifiers to other types. For example, gene set enrichment analysis is very common in omics downstream process...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 4 Gene ID conversion | Genekitr: Gene Analysis Toolkit based on R">
<meta name="twitter:description" content="If you have handled genomics analysis, then you may spend much time converting gene identifiers to other types. For example, gene set enrichment analysis is very common in omics downstream process...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.22/datatables.js"></script><link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Genekitr: Gene Analysis Toolkit based on R</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">üß¨ Welcome to genekitr!</a></li>
<li><a class="" href="quick-start.html"><span class="header-section-number">1</span> Quick start</a></li>
<li class="book-part">Part I: Search Information</li>
<li><a class="" href="search-gene-metadata-1.html"><span class="header-section-number">2</span> Search gene metadata</a></li>
<li><a class="" href="search-pubmed-1.html"><span class="header-section-number">3</span> Search PubMed records</a></li>
<li class="book-part">Part II: ID Conversion</li>
<li><a class="active" href="gene-id-conversion-1.html"><span class="header-section-number">4</span> Gene ID conversion</a></li>
<li><a class="" href="protein-id-conversion-1.html"><span class="header-section-number">5</span> Protein ID conversion</a></li>
<li><a class="" href="probe-id-conversion-1.html"><span class="header-section-number">6</span> Probe ID conversion</a></li>
<li class="book-part">Part III: Enrichment Analysis</li>
<li><a class="" href="ora-go-analysis-1.html"><span class="header-section-number">7</span> ORA GO analysis</a></li>
<li><a class="" href="ora-kegg-analysis-1.html"><span class="header-section-number">8</span> ORA KEGG analysis</a></li>
<li><a class="" href="functional-class-scoring-1.html"><span class="header-section-number">9</span> Functional Class Scoring</a></li>
<li class="book-part">Part IV: Plotting</li>
<li><a class="" href="plot-go-kegg-1.html"><span class="header-section-number">10</span> Plot GO and KEGG</a></li>
<li><a class="" href="plot-gsea-1.html"><span class="header-section-number">11</span> Plot GSEA</a></li>
<li><a class="" href="plot-others-1.html"><span class="header-section-number">12</span> Plot others</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="gene-id-conversion-1" class="section level1">
<h1>
<span class="header-section-number">4</span> Gene ID conversion<a class="anchor" aria-label="anchor" href="#gene-id-conversion-1"><i class="fas fa-link"></i></a>
</h1>
<p>If you have handled genomics analysis, then you may spend much time converting gene identifiers to other types.</p>
<blockquote>
<p>For example, gene set enrichment analysis is very common in omics downstream process which is mainly based on MsigDb. According to its <a href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/MSigDB_v7.0_Release_Notes">release note</a>, <strong>since MSigDB 7.0, identifiers for genes are mapped to their HGNC approved Gene Symbol and NCBI Gene ID through annotations extracted from Ensembl‚Äôs BioMart data service</strong>, and will be updated at each MSigDB release with the latest available version of Ensembl. Currently, MSigDB 7.5 has updated human gene annotations to Ensembl v105.</p>
</blockquote>
<p>Hence, id conversion could help keeping update with the latest annotations.</p>
<p>The most common gene id types are <strong>symbol</strong> (e.g.¬†TP53),<strong>NCBI Entrez</strong> (e.g.¬†7157) and <strong>Ensembl</strong> (e.g.¬†ENSG00000141510). ID conversion among different types is very common and there are two main concerns: performance and flexibility.</p>
<div id="convertion-current-tools" class="section level2">
<h2>
<span class="header-section-number">4.1</span> Current tools<a class="anchor" aria-label="anchor" href="#convertion-current-tools"><i class="fas fa-link"></i></a>
</h2>
<p>Currently various tools are available, however different methods has their drawbacks:</p>
<ul>
<li>
<a href="http://bioconductor.org/packages/biomaRt">biomaRt</a>: The R package is frequently used to access up-to-date Ensembl database. It has to access online database each time and built-in parameters are too many.</li>
<li>
<a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>: The R package depends on Bioconductor annotation packages to extract NCBI gene. NCBI database updates every day while Bioconductor package updates nearly every six months. Besides, it only supports 20 organisms and user should download and load annotation data before mapping.</li>
<li>
<a href="https://biit.cs.ut.ee/gprofiler/convert">gProfiler</a>: The web server supports 40 types of IDs for more than 60 species. But it only supports one-to-one conversion, for example, if user want to convert gene symbol to both Entrez and Ensembl, it could not happen. Besides, the output result is not convenient to further process.</li>
<li>
<a href="https://david.ncifcrf.gov/conversion.jsp">DAVID</a>: The web server interface is not user-friendly.</li>
<li>
<a href="https://www.uniprot.org/uploadlists/">UniProt</a>: The large protein database may support almost all species but user needs to specify input id type and only converts to protein id.</li>
<li>
<a href="http://resource.ibab.ac.in/GIDCON/geneid/home.html">GIDcon</a>: The web server only covers human, mouse and rat. User must seperates gene ids with comma if does batch query. Five symbols will take 8 secs to convert.</li>
</ul>
<p>Considered above problems, <strong>genekitr combines both Ensembl and NCBI database</strong> to provide more comprehensive result while optimizes running speed.</p>
<p>Also it supports <strong>195 vertebrate species, 120 plant species and 2 bacteria species</strong> (see also <a href="search-gene-metadata-1.html#geninfo-supported-species">session 1.1</a>).</p>
</div>
<div id="transid-basic-usage" class="section level2">
<h2>
<span class="header-section-number">4.2</span> Basic usage<a class="anchor" aria-label="anchor" href="#transid-basic-usage"><i class="fas fa-link"></i></a>
</h2>
<p><code>transId</code> provides five arguments:</p>
<ul>
<li><p><code>id</code>: gene id (symbol, Entrez or Ensembl), protein id (see also <a href="##protein-id-conversion">session 4.2</a>) or microarray id (see also <a href="##probe-id-conversion">session 5.1</a>)</p></li>
<li><p><code>transTo</code>: the conversion target type. User could select <strong>more than one</strong> from ‚Äúsymbol‚Äù, ‚Äúentrez‚Äù, ‚Äúensembl‚Äù or ‚Äúuniprot‚Äù. Besides, abbreviation is acceptable (e.g.¬†‚Äúens‚Äù is fine for ‚Äúensembl‚Äù but ‚Äúen‚Äù will raise error because it could not separate from ‚Äúentrez‚Äù)</p></li>
<li><p><code>org</code>: organism name, default is human</p></li>
<li><p><code>unique</code>: <code>TRUE</code> or <code>FALSE</code> (default). If <code>TRUE</code>, only returns one matched id.</p></li>
<li><p><code>keepNA</code>: <code>TRUE</code> or <code>FALSE</code> (default). If some id has no matched one, keep it or not.</p></li>
</ul>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># human genes</span>
<span class="va">id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"AKT3"</span>, <span class="st">"SSX6P"</span>, <span class="st">"FAKE_ID"</span>, <span class="st">"BCC7"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/transId.html">transId</a></span><span class="op">(</span><span class="va">id</span>, transTo <span class="op">=</span> <span class="st">"ens"</span><span class="op">)</span></code></pre></div>
<pre><code>##   input_id         ensembl
## 1     AKT3 ENSG00000275199
## 2     AKT3 ENSG00000117020
## 3    SSX6P ENSG00000171483
## 4     BCC7 ENSG00000141510</code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># fruit fly genes</span>
<span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/transId.html">transId</a></span><span class="op">(</span>
  id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"10178780"</span>, <span class="st">"10178777"</span>, <span class="st">"10178786"</span>, <span class="st">"10178792"</span><span class="op">)</span>,
  transTo <span class="op">=</span> <span class="st">"sym"</span>, org <span class="op">=</span> <span class="st">"fly"</span>
<span class="op">)</span></code></pre></div>
<pre><code>##   input_id          symbol
## 1 10178780         CR33929
## 2 10178777         CG42835
## 3 10178786 Su(Ste):CR40820
## 4 10178792         CG42694</code></pre>
<p>See the human example above, due to NA was removed, the result order is slightly different from input id.</p>
<p>If you want to <strong>get the same order</strong>, you can set <code>keepNA = TRUE</code>:</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/transId.html">transId</a></span><span class="op">(</span><span class="va">id</span>, transTo <span class="op">=</span> <span class="st">"ens"</span>, keepNA <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>##   input_id         ensembl
## 1     AKT3 ENSG00000275199
## 2     AKT3 ENSG00000117020
## 3    SSX6P ENSG00000171483
## 4  FAKE_ID            &lt;NA&gt;
## 5     BCC7 ENSG00000141510</code></pre>
<p>If you want to <strong>get one-to-one mapping</strong>, you can set <code>unique = TRUE</code> and the result with more comprehensive information will return:</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/transId.html">transId</a></span><span class="op">(</span><span class="va">id</span>, transTo <span class="op">=</span> <span class="st">"ens"</span>, keepNA <span class="op">=</span> <span class="cn">TRUE</span>, unique <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>##    input_id         ensembl
## 21     AKT3 ENSG00000117020
## 1     SSX6P ENSG00000171483
## 2   FAKE_ID            &lt;NA&gt;
## 3      BCC7 ENSG00000141510</code></pre>
</div>
<div id="transid-features" class="section level2">
<h2>
<span class="header-section-number">4.3</span> Features<a class="anchor" aria-label="anchor" href="#transid-features"><i class="fas fa-link"></i></a>
</h2>
<div id="transid-feature-1" class="section level3">
<h3>
<span class="header-section-number">4.3.1</span> f1: convert to several types<a class="anchor" aria-label="anchor" href="#transid-feature-1"><i class="fas fa-link"></i></a>
</h3>
<p><code>transTo</code> could accepts more than one type:</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/transId.html">transId</a></span><span class="op">(</span><span class="va">id</span>, transTo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ens"</span>, <span class="st">"ent"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##   input_id         ensembl entrezid
## 1     AKT3 ENSG00000275199    10000
## 2     AKT3 ENSG00000117020    10000
## 3    SSX6P ENSG00000171483   280657
## 4     BCC7 ENSG00000141510     7157</code></pre>
</div>
<div id="transid-feature-2" class="section level3">
<h3>
<span class="header-section-number">4.3.2</span> f2: distinguish from symbol and alias<a class="anchor" aria-label="anchor" href="#transid-feature-2"><i class="fas fa-link"></i></a>
</h3>
<p>One big common problem for <a href="gene-id-conversion-1.html#convertion-current-tools">current tools</a> is confusion of gene symbol and gene alias.</p>
<p>Gene alias is an alternative name of official gene symbol, like ‚ÄúBCC7‚Äù is actually alias for ‚ÄúTP53‚Äù. Sometimes gene alias is more widely used than official symbol. For example, ‚ÄúPDL1‚Äù is the alias for ‚ÄúCD274‚Äù in fact.</p>
<p>Due to combination of NCBI and Ensembl metadata, <code>transId</code> could easily recognize gene symbol and alias by setting <code>transTo = 'symbol'</code>.</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/transId.html">transId</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"BCC7"</span>, <span class="st">"TP53"</span>, <span class="st">"PD1"</span>, <span class="st">"PDL1"</span>, <span class="st">"TET2"</span><span class="op">)</span>, <span class="st">"sym"</span><span class="op">)</span></code></pre></div>
<pre><code>##   input_id symbol
## 1     BCC7   TP53
## 2     TP53   TP53
## 3      PD1  PDCD1
## 4      PD1   SNCA
## 5      PD1 SPATA2
## 6     PDL1  CD274
## 7     TET2   TET2</code></pre>
</div>
<div id="transid-feature-3" class="section level3">
<h3>
<span class="header-section-number">4.3.3</span> f3: no worry for Ensembl version<a class="anchor" aria-label="anchor" href="#transid-feature-3"><i class="fas fa-link"></i></a>
</h3>
<p>An Ensembl stable ID consists of five parts: ENS(species)(object type)(identifier). (version)
For example, <code>ENSG00000141510.11</code> is TP53 while the version <code>.11</code> may confuse many tools.
However using <code>transId</code>, user will not worry this issue any more.</p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/transId.html">transId</a></span><span class="op">(</span><span class="st">"ENSG00000141510.11"</span>, <span class="st">"symbol"</span><span class="op">)</span></code></pre></div>
<pre><code>##          input_id symbol
## 1 ENSG00000141510   TP53</code></pre>
</div>
</div>
<div id="transid-ex1-data" class="section level2">
<h2>
<span class="header-section-number">4.4</span> Excercise 1: DEG result<a class="anchor" aria-label="anchor" href="#transid-ex1-data"><i class="fas fa-link"></i></a>
</h2>
<p>The built-in example data is differentially expressed genes from: <a href="https://www.ncbi.nlm.nih.gov//geo/query/acc.cgi?acc=GSE42872">GSE42872</a></p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">deg</span>, package <span class="op">=</span> <span class="st">"genekitr"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">deg</span><span class="op">$</span><span class="va">change</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   down stable     up 
##    451  17009    579</code></pre>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">id</span> <span class="op">&lt;-</span> <span class="va">deg</span><span class="op">$</span><span class="va">entrezid</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 18039</code></pre>
<div id="transid-ex1-bitr" class="section level3">
<h3>
<span class="header-section-number">4.4.1</span> <code>clusterProfiler::bitr</code><a class="anchor" aria-label="anchor" href="#transid-ex1-bitr"><i class="fas fa-link"></i></a>
</h3>
<p>User needs to specify input id type and converted type from <code>columns(org.Hs.eg.db)</code>.</p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># User must firstly install the org.Hs.eg.db</span>
<span class="co"># BiocManager::install('org.Hs.eg.db')</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler</a></span><span class="op">)</span>
<span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">columns</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "ACCNUM"       "ALIAS"        "ENSEMBL"      "ENSEMBLPROT"  "ENSEMBLTRANS"
##  [6] "ENTREZID"     "ENZYME"       "EVIDENCE"     "EVIDENCEALL"  "GENENAME"    
## [11] "GO"           "GOALL"        "IPI"          "MAP"          "OMIM"        
## [16] "ONTOLOGY"     "ONTOLOGYALL"  "PATH"         "PFAM"         "PMID"        
## [21] "PROSITE"      "REFSEQ"       "SYMBOL"       "UCSCKG"       "UNIGENE"     
## [26] "UNIPROT"</code></pre>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bitr_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/bitr.html">bitr</a></span><span class="op">(</span><span class="va">id</span>, fromType <span class="op">=</span> <span class="st">"ENTREZID"</span>, toType <span class="op">=</span> <span class="st">"SYMBOL"</span>, OrgDb <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span><span class="op">)</span>
<span class="va">bitr_sym</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">bitr_res</span><span class="op">$</span><span class="va">SYMBOL</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">bitr_sym</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 18036</code></pre>
</div>
<div id="transid-ex1-biomart" class="section level3">
<h3>
<span class="header-section-number">4.4.2</span> <code>biomaRt::getBM</code><a class="anchor" aria-label="anchor" href="#transid-ex1-biomart"><i class="fas fa-link"></i></a>
</h3>
<p>The initializing process depends mainly on Ensembl‚Äôs web access speed. Usually, it may take tens of seconds.</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># initializing...</span>
<span class="va">ensembl</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useEnsembl.html">useEnsembl</a></span><span class="op">(</span>
  biomart <span class="op">=</span> <span class="st">"genes"</span>,
  dataset <span class="op">=</span> <span class="st">"hsapiens_gene_ensembl"</span>
<span class="op">)</span>
<span class="co"># conversion</span>
<span class="va">biomart_res</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/getBM.html">getBM</a></span><span class="op">(</span>
  attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"entrezgene_id"</span>, <span class="st">"external_gene_name"</span><span class="op">)</span>,
  filters <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"entrezgene_id"</span><span class="op">)</span>,
  mart <span class="op">=</span> <span class="va">ensembl</span>,
  values <span class="op">=</span> <span class="va">id</span>
<span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">biomart_sym</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">biomart_res</span><span class="op">$</span><span class="va">external_gene_name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu">stringi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/stringi/man/stri_remove_empty.html">stri_remove_empty</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">biomart_sym</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 16085</code></pre>
</div>
<div id="transid-ex1-transid" class="section level3">
<h3>
<span class="header-section-number">4.4.3</span> <code>genekitr::transId</code><a class="anchor" aria-label="anchor" href="#transid-ex1-transid"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">transid_res</span> <span class="op">&lt;-</span> <span class="fu">genekitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/transId.html">transId</a></span><span class="op">(</span><span class="va">id</span>, transTo <span class="op">=</span> <span class="st">"sym"</span>, unique <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="va">transid_sym</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">transid_res</span><span class="op">$</span><span class="va">symbol</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">transid_sym</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 18039</code></pre>
</div>
<div id="transid-ex1-compare" class="section level3">
<h3>
<span class="header-section-number">4.4.4</span> compare three results<a class="anchor" aria-label="anchor" href="#transid-ex1-compare"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/plotVenn.html">plotVenn</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
  bitr <span class="op">=</span> <span class="va">bitr_sym</span>,
  transid <span class="op">=</span> <span class="va">transid_sym</span>,
  biomart <span class="op">=</span> <span class="va">biomart_sym</span>
<span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="genekitr_files/figure-html/unnamed-chunk-45-1.png" width="98%"></div>
<div id="bitr-vs-transid" class="section level4">
<h4>
<span class="header-section-number">4.4.4.1</span> <strong><code>bitr</code> vs <code>transId</code></strong><a class="anchor" aria-label="anchor" href="#bitr-vs-transid"><i class="fas fa-link"></i></a>
</h4>
<p>There are 117 genes only in <code>bitr</code> result:</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bitr_sym</span><span class="op">[</span><span class="op">!</span><span class="va">bitr_sym</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">transid_sym</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "C19orf71"     "LINC00476"    "LOC100129484" "PP2672"       "LOC100130449"</code></pre>
<p>Let‚Äôs take a look for the first one LINC00337, which is converted from gene <a href="https://www.ncbi.nlm.nih.gov/gene/148645">148645</a></p>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bitr_res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">SYMBOL</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="st">"LINC00337"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu">pull</span><span class="op">(</span><span class="va">ENTREZID</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "148645"</code></pre>
<p>When we search it on NCBI and we could see the official symbol is ‚ÄúICMT-DT‚Äù now while ‚ÄúLINC00337‚Äù is an alias. When we look back into <code>transId</code> result, it keeps update with NCBI web.</p>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">transid_res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">input_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="st">"148645"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "ICMT-DT"</code></pre>
<p>Besides, there are 118 unique symbols in <code>transId</code> result:</p>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">transid_sym</span><span class="op">[</span><span class="op">!</span><span class="va">transid_sym</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">bitr_sym</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "TEKTIP1"     "ERCC6L2-AS1" "LINC02983"   "LINC02961"   "GPC1-AS1"</code></pre>
<p>The above gene <code>ICMT-DT</code> is listed.</p>
</div>
<div id="biomart-vs-transid" class="section level4">
<h4>
<span class="header-section-number">4.4.4.2</span> <strong><code>biomart</code> vs <code>transId</code></strong><a class="anchor" aria-label="anchor" href="#biomart-vs-transid"><i class="fas fa-link"></i></a>
</h4>
<p>There are 13 genes only in <code>biomart</code> result:</p>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">biomart_sym</span><span class="op">[</span><span class="op">!</span><span class="va">biomart_sym</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">transid_sym</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "KLRK1"     "FLJ37453"  "C11orf49"  "C19orf71"  "ZNF761"    "LINC01588"</code></pre>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">biomart_res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">external_gene_name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="st">"KLRK1"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu">pull</span><span class="op">(</span><span class="va">entrezgene_id</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 100528032</code></pre>
<p>When we search NCBI <a href="https://www.ncbi.nlm.nih.gov/gene/100528032">100528032</a> and we could see the official symbol is ‚ÄúKLRC4-KLRK1‚Äù. When we look back into <code>transId</code> and <code>bitr</code> result, they are the same with NCBI web.</p>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bitr_res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">ENTREZID</span> <span class="op">==</span> <span class="st">"100528032"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu">pull</span><span class="op">(</span><span class="va">SYMBOL</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "KLRC4-KLRK1"</code></pre>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">transid_res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">input_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="st">"100528032"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu">pull</span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "KLRC4-KLRK1"</code></pre>
</div>
</div>
</div>
<div id="transid-ex2-data" class="section level2">
<h2>
<span class="header-section-number">4.5</span> Excercise 2: mixture of gene symbol and alias<a class="anchor" aria-label="anchor" href="#transid-ex2-data"><i class="fas fa-link"></i></a>
</h2>
<p>Compared with Entrez and Ensembl, gene symbol and alias is hard to distinguish. In real science life, the impact of misunderstanding both names is huge.</p>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/29731861/">Takehara et al.¬†(2018)</a> had to retract the published paper by its authors just because they misused ‚ÄúTAZ‚Äù as the actual research target ‚ÄúWWTR1‚Äù in which ‚ÄúTAZ‚Äù is an alias of ‚ÄúWWTR1‚Äù.</p>
<p>The similar issues have published in paper: <a href="https://www.sciencedirect.com/science/article/pii/S0002929721003402">The risks of using unapproved gene symbols</a>. Of course, it is not only happens in human research. So we must raise alert of gene alias.</p>
<div id="hgnchelper-vs-transid" class="section level3">
<h3>
<span class="header-section-number">4.5.1</span> <strong><code>HGNChelper</code> vs <code>transId</code></strong><a class="anchor" aria-label="anchor" href="#hgnchelper-vs-transid"><i class="fas fa-link"></i></a>
</h3>
<p><a href="https://CRAN.R-project.org/package=HGNChelper">HGNChelper</a> is mainly built for human research, so here we will give simple human gene list as example.</p>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TBET"</span>, <span class="st">"B220"</span>, <span class="st">"BCC7"</span>, <span class="st">"PD1"</span>, <span class="st">"PDL1"</span><span class="op">)</span>

<span class="va">transid_res</span> <span class="op">&lt;-</span> <span class="fu">genekitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/transId.html">transId</a></span><span class="op">(</span><span class="va">id</span>, transTo <span class="op">=</span> <span class="st">"sym"</span>, unique <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">transid_res</span></code></pre></div>
<pre><code>##   input_id symbol
## 1     TBET  TBX21
## 2     B220  PTPRC
## 3     BCC7   TP53
## 4      PD1  PDCD1
## 5      PD1   SNCA
## 6      PD1 SPATA2
## 7     PDL1  CD274</code></pre>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hgnc_res</span> <span class="op">&lt;-</span> <span class="fu">HGNChelper</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HGNChelper/man/checkGeneSymbols.html">checkGeneSymbols</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span>
<span class="va">hgnc_res</span></code></pre></div>
<pre><code>##      x Approved          Suggested.Symbol
## 1 TBET    FALSE                      &lt;NA&gt;
## 2 B220    FALSE                      &lt;NA&gt;
## 3 BCC7    FALSE                      &lt;NA&gt;
## 4  PD1    FALSE PDCD1 /// SNCA /// SPATA2
## 5 PDL1    FALSE                     CD274</code></pre>
<p>The first three genes are omitted in <code>HGNChelper</code>.</p>
</div>
<div id="genesymbolthesarus-vs-transid" class="section level3">
<h3>
<span class="header-section-number">4.5.2</span> <strong><code>GeneSymbolThesarus</code> vs <code>transId</code></strong><a class="anchor" aria-label="anchor" href="#genesymbolthesarus-vs-transid"><i class="fas fa-link"></i></a>
</h3>
<p><code>GeneSymbolThesarus</code> is a function of <a href="https://CRAN.R-project.org/package=Seurat">Seurat</a> which finds current gene symbols from the HUGO Gene Nomenclature Committee (HGNC). It will search online so the speed will not fast.</p>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span>
<span class="va">start</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html">Sys.time</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">seurat_res</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/UpdateSymbolList.html">GeneSymbolThesarus</a></span><span class="op">(</span>symbols <span class="op">=</span> <span class="va">id</span><span class="op">)</span>
<span class="va">end</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html">Sys.time</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">(</span><span class="va">end</span> <span class="op">-</span> <span class="va">start</span><span class="op">)</span></code></pre></div>
<pre><code>## Time difference of 57.67489 secs</code></pre>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat_res</span></code></pre></div>
<pre><code>##    PDL1 
## "CD274"</code></pre>
<p>Seurat only returns one records.</p>
</div>
<div id="alias2symbol-vs-transid" class="section level3">
<h3>
<span class="header-section-number">4.5.3</span> <strong><code>alias2Symbol</code> vs <code>transId</code></strong><a class="anchor" aria-label="anchor" href="#alias2symbol-vs-transid"><i class="fas fa-link"></i></a>
</h3>
<p><code>alias2Symbol</code> is a function of <a href="http://bioconductor.org/packages/limma">limma</a> which depends on organism annotation package of Bioconductor.</p>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">limma_res</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/alias2Symbol.html">alias2Symbol</a></span><span class="op">(</span><span class="va">id</span>, species <span class="op">=</span> <span class="st">"Hs"</span><span class="op">)</span>
<span class="co"># compare with transId result</span>
<span class="va">id</span></code></pre></div>
<pre><code>## [1] "TBET" "B220" "BCC7" "PD1"  "PDL1"</code></pre>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">limma_res</span></code></pre></div>
<pre><code>## [1] "CD274"  "TBX21"  "PDCD1"  "PTPRC"  "SNCA"   "TP53"   "SPATA2"</code></pre>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">transid_res</span><span class="op">$</span><span class="va">symbol</span></code></pre></div>
<pre><code>## [1] "TBX21"  "PTPRC"  "TP53"   "PDCD1"  "SNCA"   "SPATA2" "CD274"</code></pre>
<p>It seems that <code>alias2Symbol</code> result remains disorder from input id, which may cause misunderstanding for users, while <code>transId</code> keeps the original order (if user want to get one-to-one match, just pass <code>unique = TRUE</code> to function).</p>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="search-pubmed-1.html"><span class="header-section-number">3</span> Search PubMed records</a></div>
<div class="next"><a href="protein-id-conversion-1.html"><span class="header-section-number">5</span> Protein ID conversion</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#gene-id-conversion-1"><span class="header-section-number">4</span> Gene ID conversion</a></li>
<li><a class="nav-link" href="#convertion-current-tools"><span class="header-section-number">4.1</span> Current tools</a></li>
<li><a class="nav-link" href="#transid-basic-usage"><span class="header-section-number">4.2</span> Basic usage</a></li>
<li>
<a class="nav-link" href="#transid-features"><span class="header-section-number">4.3</span> Features</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#transid-feature-1"><span class="header-section-number">4.3.1</span> f1: convert to several types</a></li>
<li><a class="nav-link" href="#transid-feature-2"><span class="header-section-number">4.3.2</span> f2: distinguish from symbol and alias</a></li>
<li><a class="nav-link" href="#transid-feature-3"><span class="header-section-number">4.3.3</span> f3: no worry for Ensembl version</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#transid-ex1-data"><span class="header-section-number">4.4</span> Excercise 1: DEG result</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#transid-ex1-bitr"><span class="header-section-number">4.4.1</span> clusterProfiler::bitr</a></li>
<li><a class="nav-link" href="#transid-ex1-biomart"><span class="header-section-number">4.4.2</span> biomaRt::getBM</a></li>
<li><a class="nav-link" href="#transid-ex1-transid"><span class="header-section-number">4.4.3</span> genekitr::transId</a></li>
<li><a class="nav-link" href="#transid-ex1-compare"><span class="header-section-number">4.4.4</span> compare three results</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#transid-ex2-data"><span class="header-section-number">4.5</span> Excercise 2: mixture of gene symbol and alias</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#hgnchelper-vs-transid"><span class="header-section-number">4.5.1</span> HGNChelper vs transId</a></li>
<li><a class="nav-link" href="#genesymbolthesarus-vs-transid"><span class="header-section-number">4.5.2</span> GeneSymbolThesarus vs transId</a></li>
<li><a class="nav-link" href="#alias2symbol-vs-transid"><span class="header-section-number">4.5.3</span> alias2Symbol vs transId</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Genekitr: Gene Analysis Toolkit based on R</strong>" was written by Yunze Liu. It was last built on 2022-06-26.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
