<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 7 ORA GO analysis | Genekitr: Gene Analysis Toolkit based on R</title>
<meta name="author" content="Yunze Liu">
<meta name="description" content="As the paper ‚ÄúUrgent need for consistent standards in functional enrichment analysis‚Äù mentioned, popular functional enrichment tools can be classified into two main categories: (1)...">
<meta name="generator" content="bookdown 0.26 with bs4_book()">
<meta property="og:title" content="Chapter 7 ORA GO analysis | Genekitr: Gene Analysis Toolkit based on R">
<meta property="og:type" content="book">
<meta property="og:description" content="As the paper ‚ÄúUrgent need for consistent standards in functional enrichment analysis‚Äù mentioned, popular functional enrichment tools can be classified into two main categories: (1)...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 7 ORA GO analysis | Genekitr: Gene Analysis Toolkit based on R">
<meta name="twitter:description" content="As the paper ‚ÄúUrgent need for consistent standards in functional enrichment analysis‚Äù mentioned, popular functional enrichment tools can be classified into two main categories: (1)...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.22/datatables.js"></script><link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Genekitr: Gene Analysis Toolkit based on R</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">üß¨ Welcome to genekitr!</a></li>
<li><a class="" href="quick-start.html"><span class="header-section-number">1</span> Quick start</a></li>
<li class="book-part">Part I: Search Information</li>
<li><a class="" href="search-gene-metadata-1.html"><span class="header-section-number">2</span> Search gene metadata</a></li>
<li><a class="" href="search-pubmed-1.html"><span class="header-section-number">3</span> Search PubMed records</a></li>
<li class="book-part">Part II: ID Conversion</li>
<li><a class="" href="gene-id-conversion-1.html"><span class="header-section-number">4</span> Gene ID conversion</a></li>
<li><a class="" href="protein-id-conversion-1.html"><span class="header-section-number">5</span> Protein ID conversion</a></li>
<li><a class="" href="probe-id-conversion-1.html"><span class="header-section-number">6</span> Probe ID conversion</a></li>
<li class="book-part">Part III: Enrichment Analysis</li>
<li><a class="active" href="ora-go-analysis-1.html"><span class="header-section-number">7</span> ORA GO analysis</a></li>
<li><a class="" href="ora-kegg-analysis-1.html"><span class="header-section-number">8</span> ORA KEGG analysis</a></li>
<li><a class="" href="functional-class-scoring-1.html"><span class="header-section-number">9</span> Functional Class Scoring</a></li>
<li><a class="" href="import-external-data.html"><span class="header-section-number">10</span> Import External Data</a></li>
<li class="book-part">Part IV: Plotting</li>
<li><a class="" href="plot-go-kegg-1.html"><span class="header-section-number">11</span> Plot GO and KEGG</a></li>
<li><a class="" href="plot-gsea-1.html"><span class="header-section-number">12</span> Plot GSEA</a></li>
<li><a class="" href="plot-others-1.html"><span class="header-section-number">13</span> Plot others</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="ora-go-analysis-1" class="section level1">
<h1>
<span class="header-section-number">7</span> ORA GO analysis<a class="anchor" aria-label="anchor" href="#ora-go-analysis-1"><i class="fas fa-link"></i></a>
</h1>
<p>As the paper <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1009935">‚ÄúUrgent need for consistent standards in functional enrichment analysis‚Äù</a> mentioned, popular functional enrichment tools can be classified into two main categories: (1) over-representation analysis (ORA) and (2) functional class scoring (FCS) .</p>
<p>According to paper <a href="https://academic.oup.com/bioinformatics/article/31/1/102/2365686?login=true">‚ÄúOver-representation of correlation analysis (ORCA): a method for identifying associations between variable sets‚Äù</a>:</p>
<blockquote>
<p>Over-representation analysis (ORA) is a simple method for objectively deciding whether a set of variables of known or suspected biological relevance, such as a gene set or pathway, is more prevalent in a set of variables of interest than we expect by chance.</p>
</blockquote>
<p>ORA usually has these procedures:</p>
<ol style="list-style-type: decimal">
<li>Identify features, such as mRNA, proteins or microarray probes, considered ‚Äúdifferentially expressed‚Äù in two or more conditions</li>
<li>Determine the number of differentially expressed features in each pathway</li>
<li>For each pathway, using a hypergeometric distribution to calculate a probability value ( P -value) of obtaining the number of differentially expressed features against the background list of all features</li>
</ol>
<div id="go-introduction" class="section level2">
<h2>
<span class="header-section-number">7.1</span> GO intruduction<a class="anchor" aria-label="anchor" href="#go-introduction"><i class="fas fa-link"></i></a>
</h2>
<blockquote>
<p>According to Wikipedia, ‚ÄúOntologies consist of detectable or directly observable representations of things and the relationships between those things.‚Äù</p>
</blockquote>
<p>GO is short for <a href="http://geneontology.org">Gene Ontology</a>. GO analysis is to find the associations between gene products and GO termsÔºå which has three domains:</p>
<ul>
<li>Biological Processes (BP)
<ul>
<li>A biological process represents a specific objective that the organism is genetically programmed to achieve.</li>
</ul>
</li>
<li>Molecular Functions (MF)
<ul>
<li>A molecular process that can be carried out by the action of a single macromolecular machine, usually via direct physical interactions with other molecular entities.</li>
</ul>
</li>
<li>Cellular Components (CC)
<ul>
<li>A location, relative to cellular compartments and structures, occupied by a macromolecular machine when it carries out a molecular function.</li>
</ul>
</li>
</ul>
<p>GO terms are built in a directed acyclic graph with a parent-child relationship.</p>
<blockquote>
<p>For more comprehensive introduction of GO, you may visit: <a href="https://advaitabio.com/faq-items/understanding-gene-ontology/" class="uri">https://advaitabio.com/faq-items/understanding-gene-ontology/</a> OR <a href="http://geneontology.org/docs/ontology-documentation/" class="uri">http://geneontology.org/docs/ontology-documentation/</a></p>
</blockquote>
</div>
<div id="go-basic-usage" class="section level2">
<h2>
<span class="header-section-number">7.2</span> Basic usage<a class="anchor" aria-label="anchor" href="#go-basic-usage"><i class="fas fa-link"></i></a>
</h2>
<blockquote>
<p><code>genGO</code> is built on <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">clusterProfiler::enrichGO</a></code> and optimizes analysis process to make better understanding.</p>
</blockquote>
<p>The simplest arguments are:</p>
<ul>
<li>
<code>id</code>: gene id of entrez (numeric id is also supported), ensembl, symbol (alias is also accepted) or uniprot</li>
<li>
<code>org</code>: organism (from <code><a href="https://rdrr.io/pkg/genekitr/man/Datasets.html">genekitr::biocOrg_name</a></code>)</li>
<li>
<code>ont</code>: ontology of ‚Äúbp‚Äù, ‚Äúmf‚Äù, ‚Äúcc‚Äù or ‚Äúall‚Äù</li>
<li><code>pvalueCutoff</code></li>
<li><code>qvalueCutoff</code></li>
</ul>
<div class="sourceCode" id="cb141"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">geneList</span>, package <span class="op">=</span> <span class="st">"genekitr"</span><span class="op">)</span>
<span class="va">id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">]</span>
<span class="va">ego</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/genGO.html">genGO</a></span><span class="op">(</span><span class="va">id</span>,
  org <span class="op">=</span> <span class="st">"human"</span>,
  ont <span class="op">=</span> <span class="st">"bp"</span>,
  pvalueCutoff <span class="op">=</span> <span class="fl">0.01</span>,
  qvalueCutoff <span class="op">=</span> <span class="fl">0.01</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ego</span><span class="op">)</span></code></pre></div>
<pre><code>##              Hs_BP_ID
## GO:0050673 GO:0050673
## GO:0006260 GO:0006260
## GO:0050678 GO:0050678
## GO:0010165 GO:0010165
## GO:1901992 GO:1901992
## GO:1901989 GO:1901989
##                                                           Description
## GO:0050673                              epithelial cell proliferation
## GO:0006260                                            DNA replication
## GO:0050678                regulation of epithelial cell proliferation
## GO:0010165                                          response to X-ray
## GO:1901992 positive regulation of mitotic cell cycle phase transition
## GO:1901989         positive regulation of cell cycle phase transition
##             GeneRatio   BgRatio       pvalue     p.adjust       qvalue
## GO:0050673 0.13821138 453/18866 6.263837e-09 1.770787e-05 1.356945e-05
## GO:0006260 0.10569106 273/18866 2.730876e-08 3.860093e-05 2.957970e-05
## GO:0050678 0.11382114 395/18866 3.012571e-07 2.838847e-04 2.175394e-04
## GO:0010165 0.04065041  32/18866 1.898488e-06 1.341756e-03 1.028181e-03
## GO:1901992 0.05691057  94/18866 2.698729e-06 1.525861e-03 1.169260e-03
## GO:1901989 0.05691057 110/18866 7.698650e-06 3.461411e-03 2.652461e-03
##                                                                                             geneID
## GO:0050673 5176/26471/3488/6591/8549/3725/6657/4208/407007/6422/29128/595/7039/7099/374/3690/10468
## GO:0006260                  3725/5427/157570/641/5888/6241/8318/9156/9134/55388/63967/51514/374393
## GO:0050678                 5176/26471/6591/3725/6657/4208/407007/6422/29128/595/7039/7099/374/3690
## GO:0010165                                                                  6422/641/5888/595/7516
## GO:1901992                                                 407007/113130/595/407006/8318/993/51514
## GO:1901989                                                 407007/113130/595/407006/8318/993/51514
##                                                                                                geneID_symbol
## GO:0050673 SERPINF1/NUPR1/IGFBP5/SNAI2/LGR5/JUN/SOX2/MEF2C/MIR222/SFRP1/UHRF1/CCND1/TGFA/TLR4/AREG/ITGB3/FST
## GO:0006260                           JUN/POLE2/ESCO2/BLM/RAD51/RRM2/CDC45/EXO1/CCNE2/MCM10/CLSPN/DTL/FAM111B
## GO:0050678                 SERPINF1/NUPR1/SNAI2/JUN/SOX2/MEF2C/MIR222/SFRP1/UHRF1/CCND1/TGFA/TLR4/AREG/ITGB3
## GO:0010165                                                                       SFRP1/BLM/RAD51/CCND1/XRCC2
## GO:1901992                                                        MIR222/CDCA5/CCND1/MIR221/CDC45/CDC25A/DTL
## GO:1901989                                                        MIR222/CDCA5/CCND1/MIR221/CDC45/CDC25A/DTL
##            Count FoldEnrich RichFactor
## GO:0050673    17   5.756062 0.03752759
## GO:0006260    13   7.303910 0.04761905
## GO:0050678    14   5.436328 0.03544304
## GO:0010165     5  23.965955 0.15625000
## GO:1901992     7  11.422072 0.07446809
## GO:1901989     7   9.760680 0.06363636</code></pre>
<p>If user has proteomics data, protein IDs could directly do GO analysis instead of gene conversion.
Here we will use data from: <a href="protein-id-conversion-1.html#protein-example-data">rat protein example data</a></p>
<div class="sourceCode" id="cb143"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pro_id</span> <span class="op">&lt;-</span> <span class="va">rat_prodata</span><span class="op">$</span><span class="va">Acc</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">pro_id</span>, <span class="fl">10</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "P27364" "P17475" "P14046" "P00507" "P15650" "P08503" "P16638" "P18484"
##  [9] "P06757" "P24483"</code></pre>
<div class="sourceCode" id="cb145"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rat_ego</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/genGO.html">genGO</a></span><span class="op">(</span><span class="va">pro_id</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span>, 
                 org <span class="op">=</span> <span class="st">'rat'</span>, 
                 ont <span class="op">=</span> <span class="st">'bp'</span>,
                 pvalueCutoff <span class="op">=</span> <span class="fl">0.01</span>, 
                 qvalueCutoff <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">rat_ego</span><span class="op">)</span></code></pre></div>
<pre><code>##              Rn_BP_ID                              Description GeneRatio
## GO:0006631 GO:0006631             fatty acid metabolic process 0.3218391
## GO:0120254 GO:0120254      olefinic compound metabolic process 0.2068966
## GO:0033559 GO:0033559 unsaturated fatty acid metabolic process 0.1954023
## GO:0001676 GO:0001676  long-chain fatty acid metabolic process 0.1839080
## GO:0009410 GO:0009410          response to xenobiotic stimulus 0.1839080
## GO:0006805 GO:0006805             xenobiotic metabolic process 0.1724138
##              BgRatio       pvalue     p.adjust       qvalue
## GO:0006631 463/18067 1.528743e-23 3.227175e-20 2.528057e-20
## GO:0120254 132/18067 1.339466e-21 1.413807e-18 1.107527e-18
## GO:0033559 138/18067 1.172308e-19 8.249144e-17 6.462093e-17
## GO:0001676 117/18067 2.677336e-19 1.412964e-16 1.106867e-16
## GO:0009410 131/18067 1.747526e-18 7.378055e-16 5.779712e-16
## GO:0006805 116/18067 9.145398e-18 3.217656e-15 2.520600e-15
##                                                                                                                                                                                                         geneID
## GO:0006631 P15650/P08503/P16638/P01015/P07872/P01026/P20816/P18125/P15149/P13107/P05179/P08683/P11510/P20814/P19225/P10633/P10634/P12939/P18886/P11030/P23457/P14604/P07896/P02692/P04905/P08010/P08009/P04906
## GO:0120254                                                                       P27364/P20816/P15149/P13107/P05179/P08683/P11510/P20814/P19225/P10633/P10634/P12939/P11715/P23457/P04905/P08010/P08009/P04906
## GO:0033559                                                                              P07872/P20816/P15149/P13107/P05179/P08683/P11510/P20814/P19225/P10633/P10634/P12939/P23457/P04905/P08010/P08009/P04906
## GO:0001676                                                                                     P15650/P20816/P15149/P13107/P05179/P08683/P11510/P20814/P19225/P10633/P10634/P12939/P04905/P08010/P08009/P04906
## GO:0009410                                                                                     P15149/P13107/P05179/P08683/P11510/P20814/P19225/P10633/P10634/P12939/P19468/P04903/P04905/P08010/P08009/P04906
## GO:0006805                                                                                            P15149/P13107/P05179/P08683/P11510/P20814/P19225/P10633/P10634/P12939/P04903/P04905/P08010/P08009/P04906
##                                                                                                                                                                                geneID_symbol
## GO:0006631 Acadl/Acadm/Acly/Agt/Acox1/C3/Cyp4a2/Cyp7a1/Cyp2a2/Cyp2b3/Cyp2c7/Cyp2c11/Cyp2c12/Cyp2c13/Cyp2c70/Cyp2d1/Cyp2d2/Cyp2d10/Cpt2/Dbi/Akr1c9/Echs1/Ehhadh/Fabp1/Gstm1/Gstm2/Gstm7/Gstp1
## GO:0120254                                                   Hsd3b5/Cyp4a2/Cyp2a2/Cyp2b3/Cyp2c7/Cyp2c11/Cyp2c12/Cyp2c13/Cyp2c70/Cyp2d1/Cyp2d2/Cyp2d10/Cyp17a1/Akr1c9/Gstm1/Gstm2/Gstm7/Gstp1
## GO:0033559                                                            Acox1/Cyp4a2/Cyp2a2/Cyp2b3/Cyp2c7/Cyp2c11/Cyp2c12/Cyp2c13/Cyp2c70/Cyp2d1/Cyp2d2/Cyp2d10/Akr1c9/Gstm1/Gstm2/Gstm7/Gstp1
## GO:0001676                                                                   Acadl/Cyp4a2/Cyp2a2/Cyp2b3/Cyp2c7/Cyp2c11/Cyp2c12/Cyp2c13/Cyp2c70/Cyp2d1/Cyp2d2/Cyp2d10/Gstm1/Gstm2/Gstm7/Gstp1
## GO:0009410                                                                     Cyp2a2/Cyp2b3/Cyp2c7/Cyp2c11/Cyp2c12/Cyp2c13/Cyp2c70/Cyp2d1/Cyp2d2/Cyp2d10/Gclc/Gsta2/Gstm1/Gstm2/Gstm7/Gstp1
## GO:0006805                                                                          Cyp2a2/Cyp2b3/Cyp2c7/Cyp2c11/Cyp2c12/Cyp2c13/Cyp2c70/Cyp2d1/Cyp2d2/Cyp2d10/Gsta2/Gstm1/Gstm2/Gstm7/Gstp1
##            Count FoldEnrich RichFactor
## GO:0006631    28   12.55868 0.06047516
## GO:0120254    18   28.31818 0.13636364
## GO:0033559    17   25.58213 0.12318841
## GO:0001676    16   28.39886 0.13675214
## GO:0009410    16   25.36387 0.12213740
## GO:0006805    15   26.85345 0.12931034</code></pre>
<p>Except for the GO enrichment result, protein IDs are also mapped to gene symbols which is easy understanding.</p>
<div id="go-additional-arguments" class="section level3">
<h3>
<span class="header-section-number">7.2.1</span> Additional arguments<a class="anchor" aria-label="anchor" href="#go-additional-arguments"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>
<code>pAdjustMethod</code>: choose from ‚Äúholm‚Äù, ‚Äúhochberg‚Äù, ‚Äúhommel‚Äù, ‚Äúbonferroni‚Äù, ‚ÄúBH‚Äù, ‚ÄúBY‚Äù, ‚Äúfdr‚Äù, ‚Äúnone‚Äù</li>
<li>
<code>minGSSize</code>: Minimal size of each gene set for analysis. Default is 10. Pathway with less than ten genes will be omitted (e.g.¬†<a href="http://www.informatics.jax.org/vocab/gene_ontology/GO:0062196">GO:0062196</a> is a child term with 8-gene in the bottom of the directed acyclic graph).</li>
<li>
<code>maxGSSize</code>: Maximal size. Default is 500. The pathway will more than 500 genes will be omitted (e.g.¬†<a href="http://www.informatics.jax.org/vocab/gene_ontology/GO:0007049">GO:0007049</a> has 1796 genes as parent nodes at the top of the directed acyclic graph).</li>
<li>
<code>universe</code>: If uses have their background genes, set them as <code>universe</code>. Otherwise, the program will take all genes in the <code>orgDb</code> annotation package (e.g.¬†<code>org.Hs.eg.db</code>).</li>
</ul>
</div>
<div id="go-new-features" class="section level3">
<h3>
<span class="header-section-number">7.2.2</span> New features<a class="anchor" aria-label="anchor" href="#go-new-features"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>If input id is not a symbol, <code>genGO</code> will convert to a symbol for easy understanding. Besides, if input id is mixture of gene symbol and alias, the result will convert alias to symbol automatically and result will show two columns <code>geneID</code> and <code>geneID_symbol</code>. <strong>If input id is symbol and the result only has <code>geneID</code>, that means the input symbol do not have any alias.</strong>
</li>
<li>No need to sepcify input id type, <code>genGO</code> will allocate automatically.</li>
<li>
<code>genGO</code> sets four measurement aspects: ‚ÄúCount‚Äù, ‚ÄúGeneRatio‚Äù, ‚ÄúFoldEnrichment‚Äù and ‚ÄúRichFactor‚Äù.
<ul>
<li>Bg Ratio: Total number of specific pathway gene/ Universal gene numbers (e.g., all human protein-coding genes)</li>
<li>Gene Ratio: Genes of interest in specific pathway/gene of interest (e.g., differentially expressed genes)</li>
<li>Fold Enrichment: GeneRatio / BgRatio</li>
<li>Rich Factor: Genes of interest in specific pathway/ Total number of specific pathway gene</li>
</ul>
</li>
<li>Return as data frame format which is easy to select and export interested pathways</li>
</ul>
</div>
</div>
<div id="go-group-enrichment" class="section level2">
<h2>
<span class="header-section-number">7.3</span> Group enrichment analysis<a class="anchor" aria-label="anchor" href="#go-group-enrichment"><i class="fas fa-link"></i></a>
</h2>
<p>If there are multi-group comparison (e.g.¬†up and down-regulated genes), user needs to specify gene groups and set them as the <code>group_list</code> argument:</p>
<div id="comparison-with-one-group" class="section level3">
<h3>
<span class="header-section-number">7.3.1</span> Comparison with one group:<a class="anchor" aria-label="anchor" href="#comparison-with-one-group"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb147"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">id2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span>, <span class="fl">50</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span>, <span class="fl">50</span><span class="op">)</span><span class="op">)</span>
<span class="va">one_group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>group1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"up"</span>, <span class="fl">50</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"down"</span>, <span class="fl">50</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">gego1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/genGO.html">genGO</a></span><span class="op">(</span><span class="va">id2</span>,
  group_list <span class="op">=</span> <span class="va">one_group</span>,
  org <span class="op">=</span> <span class="st">"human"</span>,
  ont <span class="op">=</span> <span class="st">"bp"</span>,
  pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,
  qvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gego1</span>, <span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code>##   Cluster   Hs_BP_ID                                                Description
## 1    down GO:0006260                                            DNA replication
## 2    down GO:1901992 positive regulation of mitotic cell cycle phase transition
## 3    down GO:0045931                  positive regulation of mitotic cell cycle
## 4    down GO:1901989         positive regulation of cell cycle phase transition
## 5    down GO:0010165                                          response to X-ray
##   GeneRatio   BgRatio       pvalue     p.adjust       qvalue
## 1      0.24 273/18866 4.930656e-12 8.229265e-09 5.932358e-09
## 2      0.14  94/18866 5.096508e-09 4.253036e-06 3.065952e-06
## 3      0.16 168/18866 1.306551e-08 6.402841e-06 4.615716e-06
## 4      0.14 110/18866 1.534534e-08 6.402841e-06 4.615716e-06
## 5      0.10  32/18866 2.031273e-08 6.780389e-06 4.887884e-06
##                                                              geneID
## 1 5427/157570/641/5888/6241/8318/9156/9134/55388/63967/51514/374393
## 2                           407007/113130/595/407006/8318/993/51514
## 3                      407007/113130/595/407006/7039/8318/993/51514
## 4                           407007/113130/595/407006/8318/993/51514
## 5                                            6422/641/5888/595/7516
##                                                         geneID_symbol Count
## 1 POLE2/ESCO2/BLM/RAD51/RRM2/CDC45/EXO1/CCNE2/MCM10/CLSPN/DTL/FAM111B    12
## 2                          MIR222/CDCA5/CCND1/MIR221/CDC45/CDC25A/DTL     7
## 3                     MIR222/CDCA5/CCND1/MIR221/TGFA/CDC45/CDC25A/DTL     8
## 4                          MIR222/CDCA5/CCND1/MIR221/CDC45/CDC25A/DTL     7
## 5                                         SFRP1/BLM/RAD51/CCND1/XRCC2     5
##   FoldEnrich RichFactor
## 1   16.58549 0.04395604
## 2   28.09830 0.07446809
## 3   17.96762 0.04761905
## 4   24.01127 0.06363636
## 5   58.95625 0.15625000</code></pre>
</div>
<div id="comparison-with-more-than-one-group" class="section level3">
<h3>
<span class="header-section-number">7.3.2</span> Comparison with more than one group<a class="anchor" aria-label="anchor" href="#comparison-with-more-than-one-group"><i class="fas fa-link"></i></a>
</h3>
<p>If there are two group conditions (e.g.¬†up/down regulated; treatment on serial time points), user needs to set two-group list.</p>
<div class="sourceCode" id="cb149"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">two_groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
  exp_group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"up"</span>, <span class="fl">50</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"down"</span>, <span class="fl">50</span><span class="op">)</span><span class="op">)</span>,
  time_group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"time1"</span>, <span class="fl">40</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"time2"</span>, <span class="fl">60</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">gego2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/genGO.html">genGO</a></span><span class="op">(</span><span class="va">id2</span>,
  group_list <span class="op">=</span> <span class="va">two_groups</span>,
  org <span class="op">=</span> <span class="st">"human"</span>,
  ont <span class="op">=</span> <span class="st">"bp"</span>,
  pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,
  qvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gego2</span>, <span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code>##      Cluster   Hs_BP_ID
## 1 down.time2 GO:0006260
## 2 down.time2 GO:1901992
## 3 down.time2 GO:0045931
## 4 down.time2 GO:1901989
## 5 down.time2 GO:0010165
##                                                  Description GeneRatio
## 1                                            DNA replication      0.24
## 2 positive regulation of mitotic cell cycle phase transition      0.14
## 3                  positive regulation of mitotic cell cycle      0.16
## 4         positive regulation of cell cycle phase transition      0.14
## 5                                          response to X-ray      0.10
##     BgRatio       pvalue     p.adjust       qvalue
## 1 273/18866 4.930656e-12 8.229265e-09 5.932358e-09
## 2  94/18866 5.096508e-09 4.253036e-06 3.065952e-06
## 3 168/18866 1.306551e-08 6.402841e-06 4.615716e-06
## 4 110/18866 1.534534e-08 6.402841e-06 4.615716e-06
## 5  32/18866 2.031273e-08 6.780389e-06 4.887884e-06
##                                                              geneID
## 1 5427/157570/641/5888/6241/8318/9156/9134/55388/63967/51514/374393
## 2                           407007/113130/595/407006/8318/993/51514
## 3                      407007/113130/595/407006/7039/8318/993/51514
## 4                           407007/113130/595/407006/8318/993/51514
## 5                                            6422/641/5888/595/7516
##                                                         geneID_symbol Count
## 1 POLE2/ESCO2/BLM/RAD51/RRM2/CDC45/EXO1/CCNE2/MCM10/CLSPN/DTL/FAM111B    12
## 2                          MIR222/CDCA5/CCND1/MIR221/CDC45/CDC25A/DTL     7
## 3                     MIR222/CDCA5/CCND1/MIR221/TGFA/CDC45/CDC25A/DTL     8
## 4                          MIR222/CDCA5/CCND1/MIR221/CDC45/CDC25A/DTL     7
## 5                                         SFRP1/BLM/RAD51/CCND1/XRCC2     5
##   FoldEnrich RichFactor
## 1   16.58549 0.04395604
## 2   28.09830 0.07446809
## 3   17.96762 0.04761905
## 4   24.01127 0.06363636
## 5   58.95625 0.15625000</code></pre>
</div>
</div>
<div id="go-advanced-usage" class="section level2">
<h2>
<span class="header-section-number">7.4</span> Advanced usage<a class="anchor" aria-label="anchor" href="#go-advanced-usage"><i class="fas fa-link"></i></a>
</h2>
<div id="different-types-of-input-and-backgroud-genes" class="section level3">
<h3>
<span class="header-section-number">7.4.1</span> Different types of input and backgroud genes<a class="anchor" aria-label="anchor" href="#different-types-of-input-and-backgroud-genes"><i class="fas fa-link"></i></a>
</h3>
<blockquote>
<p><code>genGO</code> could utilize <code>transId</code> to make the same gene type</p>
</blockquote>
<p>In the below example, we will set input id as symbols and background id as Entrez.</p>
<div class="sourceCode" id="cb151"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">symbol_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/transId.html">transId</a></span><span class="op">(</span><span class="va">id2</span>, transTo <span class="op">=</span> <span class="st">"symbol"</span>, unique <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/pull.html">pull</a></span><span class="op">(</span><span class="va">symbol</span><span class="op">)</span>
<span class="va">symbol_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">symbol_id</span>, <span class="st">"BCC7"</span>, <span class="st">"PD1"</span>, <span class="st">"PDL1"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">symbol_id</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "CD36"       "DCT"        "PRUNE2"     "ST6GALNAC2" "TCN1"      
## [6] "CD24"</code></pre>
<div class="sourceCode" id="cb153"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "948"       "1638"      "158471"    "10610"     "6947"      "100133941"</code></pre>
<div class="sourceCode" id="cb155"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">diy_ego</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/genGO.html">genGO</a></span><span class="op">(</span><span class="va">symbol_id</span>,
  org <span class="op">=</span> <span class="st">"human"</span>,
  ont <span class="op">=</span> <span class="st">"bp"</span>,
  pvalueCutoff <span class="op">=</span> <span class="fl">0.01</span>,
  qvalueCutoff <span class="op">=</span> <span class="fl">0.01</span>,
  pAdjustMethod <span class="op">=</span> <span class="st">"fdr"</span>,
  minGSSize <span class="op">=</span> <span class="fl">5</span>,
  maxGSSize <span class="op">=</span> <span class="fl">2000</span>,
  universe <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">geneList</span><span class="op">)</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">diy_ego</span>, <span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code>##              Hs_BP_ID                              Description  GeneRatio
## GO:0006260 GO:0006260                          DNA replication 0.12745098
## GO:0010165 GO:0010165                        response to X-ray 0.05882353
## GO:0006974 GO:0006974 cellular response to DNA damage stimulus 0.19607843
## GO:0033554 GO:0033554              cellular response to stress 0.31372549
## GO:0050673 GO:0050673            epithelial cell proliferation 0.13725490
##               BgRatio       pvalue     p.adjust       qvalue
## GO:0006260  261/16024 1.086449e-08 3.524442e-05 2.534286e-05
## GO:0010165   29/16024 2.418982e-08 3.923589e-05 2.821297e-05
## GO:0006974  801/16024 1.282570e-07 1.386886e-04 9.972543e-05
## GO:0033554 1946/16024 2.227431e-07 1.806446e-04 1.298944e-04
## GO:0050673  413/16024 3.461281e-07 2.245679e-04 1.614779e-04
##                                                                                                                                                                                                     geneID
## GO:0006260                                                                                                                        POLE2/ESCO2/BLM/RAD51/RRM2/CDC45/EXO1/CCNE2/MCM10/CLSPN/DTL/FAM111B/BCC7
## GO:0010165                                                                                                                                                                SFRP1/BLM/RAD51/CCND1/XRCC2/BCC7
## GO:0006974                                                                              NUPR1/SNAI2/UHRF1/CDCA5/PMAIP1/POLE2/ESCO2/BLM/RAD51/CCND1/MIR221/XRCC2/CDC45/EXO1/DDIAS/IER3/MCM10/CLSPN/DTL/BCC7
## GO:0033554 CD36/NUPR1/CYP1B1/SESN3/SNAI2/SLC38A3/FOLR1/CRYAB/TP53INP1/SLC1A1/MIR222/SFRP1/UHRF1/CDCA5/PMAIP1/POLE2/ESCO2/BLM/RAD51/CCND1/MIR221/XRCC2/TLR4/CDC45/EXO1/DDIAS/IER3/MCM10/CLSPN/DTL/ETV5/BCC7
## GO:0050673                                                                                                              SERPINF1/NUPR1/IGFBP5/SNAI2/LGR5/MIR222/SFRP1/UHRF1/CCND1/TGFA/TLR4/AREG/ITGB3/FST
##                                                                                                                                                                                              geneID_symbol
## GO:0006260                                                                                                                        POLE2/ESCO2/BLM/RAD51/RRM2/CDC45/EXO1/CCNE2/MCM10/CLSPN/DTL/FAM111B/TP53
## GO:0010165                                                                                                                                                                SFRP1/BLM/RAD51/CCND1/XRCC2/TP53
## GO:0006974                                                                              NUPR1/SNAI2/UHRF1/CDCA5/PMAIP1/POLE2/ESCO2/BLM/RAD51/CCND1/MIR221/XRCC2/CDC45/EXO1/DDIAS/IER3/MCM10/CLSPN/DTL/TP53
## GO:0033554 CD36/NUPR1/CYP1B1/SESN3/SNAI2/SLC38A3/FOLR1/CRYAB/TP53INP1/SLC1A1/MIR222/SFRP1/UHRF1/CDCA5/PMAIP1/POLE2/ESCO2/BLM/RAD51/CCND1/MIR221/XRCC2/TLR4/CDC45/EXO1/DDIAS/IER3/MCM10/CLSPN/DTL/ETV5/TP53
## GO:0050673                                                                                                              SERPINF1/NUPR1/IGFBP5/SNAI2/LGR5/MIR222/SFRP1/UHRF1/CCND1/TGFA/TLR4/AREG/ITGB3/FST
##            Count FoldEnrich RichFactor
## GO:0006260    13   7.824807 0.04980843
## GO:0010165     6  32.503043 0.20689655
## GO:0006974    20   3.922548 0.02496879
## GO:0033554    32   2.583318 0.01644399
## GO:0050673    14   5.325357 0.03389831</code></pre>
<p>In this result, the input <strong>gene alias is also recognized</strong> and converted to the official symbol (e.g.¬†<code>BCC7</code>):</p>
<div class="sourceCode" id="cb157"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">diy_ego</span><span class="op">$</span><span class="va">geneID</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></code></pre></div>
<pre><code>## [1] "POLE2/ESCO2/BLM/RAD51/RRM2/CDC45/EXO1/CCNE2/MCM10/CLSPN/DTL/FAM111B/BCC7"</code></pre>
<div class="sourceCode" id="cb159"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">diy_ego</span><span class="op">$</span><span class="va">geneID_symbol</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></code></pre></div>
<pre><code>## [1] "POLE2/ESCO2/BLM/RAD51/RRM2/CDC45/EXO1/CCNE2/MCM10/CLSPN/DTL/FAM111B/TP53"</code></pre>
</div>
<div id="go-simplify" class="section level3">
<h3>
<span class="header-section-number">7.4.2</span> Simplify GO result<a class="anchor" aria-label="anchor" href="#go-simplify"><i class="fas fa-link"></i></a>
</h3>
<p><code>simGO</code> utilizes <code><a href="https://rdrr.io/pkg/GOSemSim/man/goSim.html">GOSemSim::goSim()</a></code> to calculate semantic similarity between two GO terms, then passes to <code><a href="https://rdrr.io/pkg/rrvgo/man/reduceSimMatrix.html">rrvgo::reduceSimMatrix()</a></code> to reduce redundant terms.</p>
<p>Methods of calculating the similarity between nodes: ‚ÄúResnik‚Äù, ‚ÄúLin‚Äù, ‚ÄúRel‚Äù, ‚ÄúJiang‚Äù , ‚ÄúWang‚Äù and ‚ÄúJC‚Äù (Jaccard similarity coefficient).</p>
<div class="sourceCode" id="cb161"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sim_go</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/simGO.html">simGO</a></span><span class="op">(</span><span class="va">diy_ego</span>,sim_method <span class="op">=</span> <span class="st">'Wang'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">diy_ego</span><span class="op">)</span>;<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">sim_go</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 100  12</code></pre>
<pre><code>## [1] 23 12</code></pre>
<p>Let‚Äôs look which terms are considered as redundant:</p>
<div class="sourceCode" id="cb164"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">diy_ego</span><span class="op">$</span><span class="va">Hs_BP_ID</span><span class="op">[</span><span class="op">!</span><span class="va">diy_ego</span><span class="op">$</span><span class="va">Hs_BP_ID</span><span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span><span class="va">sim_go</span><span class="op">$</span><span class="va">Hs_BP_ID</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "GO:0033554" "GO:0022402" "GO:0009314" "GO:0051093" "GO:0010212"
## [6] "GO:0071479"</code></pre>


<div class="figure">
<span style="display:block;" id="fig:goSim"></span>
<img src="figures/gosim_exp.png" alt="Term GO:0006974" width="100%"><p class="caption">
Figure 7.1: <strong>Term <a href="GO:0006974" class="uri">GO:0006974</a></strong>
</p>
</div>
<p>The first record <a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006974">GO:0006974</a> is actually child node of <code>GO:0033554</code>, which is remained in <code>sim_go</code>.</p>
</div>
</div>
<div id="go-tools-comparsion" class="section level2">
<h2>
<span class="header-section-number">7.5</span> Comparison with other tools<a class="anchor" aria-label="anchor" href="#go-tools-comparsion"><i class="fas fa-link"></i></a>
</h2>
<div id="gengo-vs-enrichgo" class="section level3">
<h3>
<span class="header-section-number">7.5.1</span> <code>genGO</code> vs <code>enrichGO</code><a class="anchor" aria-label="anchor" href="#gengo-vs-enrichgo"><i class="fas fa-link"></i></a>
</h3>
<p>If input id contains gene alias, <code>genGO</code> will invoke <code>transId</code> to automatically convert gene alias to symbol for better explanation.</p>
<div class="sourceCode" id="cb166"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">check_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"BCC7"</span>, <span class="st">"PDL1"</span>, <span class="st">"PD1"</span>, <span class="st">"TET2"</span>, <span class="st">"BRCA1"</span><span class="op">)</span>

<span class="va">genekitr_res</span> <span class="op">&lt;-</span> <span class="fu">genekitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/genGO.html">genGO</a></span><span class="op">(</span><span class="va">check_gene</span>, ont <span class="op">=</span> <span class="st">"bp"</span>, org <span class="op">=</span> <span class="st">"hs"</span>, 
                                pvalueCutoff <span class="op">=</span> <span class="fl">0.01</span>, qvalueCutoff <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>

<span class="va">clustp_res</span> <span class="op">&lt;-</span> <span class="fu">clusterProfiler</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">enrichGO</a></span><span class="op">(</span><span class="va">check_gene</span>,
  OrgDb <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>,
  keyType <span class="op">=</span> <span class="st">"SYMBOL"</span>,
  pvalueCutoff <span class="op">=</span> <span class="fl">0.01</span>,
  qvalueCutoff <span class="op">=</span> <span class="fl">0.01</span>,
  ont <span class="op">=</span> <span class="st">"BP"</span>
<span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">genekitr_res</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 35 12</code></pre>
<div class="sourceCode" id="cb168"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">clustp_res</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 11  9</code></pre>
<p>If you check these two results, you may find that <code>clustp_res</code> only contains two genes: <code>TET2</code> and <code>BRCA1</code>, while <code>genekitr_res</code> includes all five genes.</p>
<div class="sourceCode" id="cb170"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">genekitr_res</span>,<span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code>##              Hs_BP_ID                                         Description
## GO:0070234 GO:0070234     positive regulation of T cell apoptotic process
## GO:0070230 GO:0070230 positive regulation of lymphocyte apoptotic process
## GO:2000108 GO:2000108  positive regulation of leukocyte apoptotic process
## GO:0070232 GO:0070232              regulation of T cell apoptotic process
## GO:0070231 GO:0070231                            T cell apoptotic process
##            GeneRatio  BgRatio       pvalue     p.adjust       qvalue
## GO:0070234       0.6 14/18866 3.250141e-09 2.141843e-06 5.234438e-07
## GO:0070230       0.6 18/18866 7.283713e-09 2.399983e-06 5.865305e-07
## GO:2000108       0.6 29/18866 3.258750e-08 7.158387e-06 1.749434e-06
## GO:0070232       0.6 34/18866 5.334592e-08 8.788740e-06 2.147875e-06
## GO:0070231       0.6 49/18866 1.640494e-07 2.162171e-05 5.284119e-06
##                   geneID    geneID_symbol Count FoldEnrich RichFactor
## GO:0070234 BCC7/PDL1/PD1 TP53/CD274/PDCD1     3   808.5429 0.21428571
## GO:0070230 BCC7/PDL1/PD1 TP53/CD274/PDCD1     3   628.8667 0.16666667
## GO:2000108 BCC7/PDL1/PD1 TP53/CD274/PDCD1     3   390.3310 0.10344828
## GO:0070232 BCC7/PDL1/PD1 TP53/CD274/PDCD1     3   332.9294 0.08823529
## GO:0070231 BCC7/PDL1/PD1 TP53/CD274/PDCD1     3   231.0122 0.06122449</code></pre>
<div class="sourceCode" id="cb172"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">clustp_res</span>,<span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code>##                    ID                      Description GeneRatio   BgRatio
## GO:0051568 GO:0051568        histone H3-K4 methylation       2/2  57/18866
## GO:0044728 GO:0044728 DNA methylation or demethylation       2/2  98/18866
## GO:0034968 GO:0034968       histone lysine methylation       2/2 116/18866
## GO:0006304 GO:0006304                 DNA modification       2/2 120/18866
## GO:0018022 GO:0018022      peptidyl-lysine methylation       2/2 132/18866
##                  pvalue    p.adjust       qvalue     geneID Count
## GO:0051568 8.968633e-06 0.001892382 9.440666e-06 TET2/BRCA1     2
## GO:0044728 2.670922e-05 0.002006407 1.000951e-05 TET2/BRCA1     2
## GO:0034968 3.748169e-05 0.002006407 1.000951e-05 TET2/BRCA1     2
## GO:0006304 4.012283e-05 0.002006407 1.000951e-05 TET2/BRCA1     2
## GO:0018022 4.858571e-05 0.002006407 1.000951e-05 TET2/BRCA1     2</code></pre>
<p>If you look into deeper, you may find six terms are unique in <code>clustp_res</code>, such as <code>GO:0016571</code>:</p>
<div class="sourceCode" id="cb174"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">clustp_res</span><span class="op">$</span><span class="va">ID</span><span class="op">[</span><span class="op">!</span><span class="va">clustp_res</span><span class="op">$</span><span class="va">ID</span><span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span><span class="va">genekitr_res</span><span class="op">$</span><span class="va">Hs_BP_ID</span> <span class="op">]</span></code></pre></div>
<pre><code>## [1] "GO:0016571" "GO:0006479" "GO:0008213" "GO:0043414" "GO:0032259"
## [6] "GO:0018205"</code></pre>
<p><strong>Question1: Why some terms are omited in <code>genGO</code> result?</strong></p>
<p>To get adjusted p value, the critical parameter is the number of testing. For simple explaination, <code>adj.p = f * p (f&gt;1)</code> and f is related with testing methods and the number of tests.</p>
<p>In the above example, <code>genGO</code> recognizes five genes so we need to do five testings while <code>enrichGO</code> includes two genes so the testing number will be less. This directly causes <code>adj.Pvalue</code> of those ‚Äúdisappeared‚Äù terms becomes higher and filtered by cutoff <code>0.01</code>.</p>
<p>If we set cutoff higher(e.g.¬†0.05), then we will still get those terms:</p>
<div class="sourceCode" id="cb176"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genekitr_res2</span> <span class="op">&lt;-</span> <span class="fu">genekitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/genGO.html">genGO</a></span><span class="op">(</span><span class="va">check_gene</span>, ont <span class="op">=</span> <span class="st">"bp"</span>, org <span class="op">=</span> <span class="st">"hs"</span>, 
                                pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>, qvalueCutoff <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">clustp_res</span><span class="op">$</span><span class="va">ID</span><span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span><span class="va">genekitr_res2</span><span class="op">$</span><span class="va">Hs_BP_ID</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## TRUE 
##   11</code></pre>
<p><strong>Question2: why <code>enrichGO</code> only recognize two genes?</strong></p>
<p>Such gene ‚ÄúBCC7‚Äù, ‚ÄúPD1‚Äù or ‚ÄúPDL1‚Äù are essential so they must have related pathways.</p>
<p>The core reason is they are actually gene aliases but <code>enrichGO</code> set argument <code>keyType = "SYMBOL"</code> which cause <code>enrichGO</code> only accepts official symbols. This function is based on organism annotation package (e.g.¬†<code>org.Hs.eg.db</code>) and it accepts different types of data:</p>
<div class="sourceCode" id="cb178"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span>
<span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">columns</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "ACCNUM"       "ALIAS"        "ENSEMBL"      "ENSEMBLPROT"  "ENSEMBLTRANS"
##  [6] "ENTREZID"     "ENZYME"       "EVIDENCE"     "EVIDENCEALL"  "GENENAME"    
## [11] "GO"           "GOALL"        "IPI"          "MAP"          "OMIM"        
## [16] "ONTOLOGY"     "ONTOLOGYALL"  "PATH"         "PFAM"         "PMID"        
## [21] "PROSITE"      "REFSEQ"       "SYMBOL"       "UCSCKG"       "UNIGENE"     
## [26] "UNIPROT"</code></pre>
<p>Only by specifying <code>keyType = 'ALIAS'</code>, this program could recognize gene alias, however it still could not distinguish the mixture of symbol and alias.</p>
</div>
<div id="gengo-vs-panther-web" class="section level3">
<h3>
<span class="header-section-number">7.5.2</span> <code>genGO</code> vs <code>Panther web</code><a class="anchor" aria-label="anchor" href="#gengo-vs-panther-web"><i class="fas fa-link"></i></a>
</h3>
<p>Still using the above five check genes.</p>
<p>Panther could recognize ‚ÄúPDL1‚Äù but not ‚ÄúPD1‚Äù, as well as ‚ÄúBCC7‚Äù is omited.</p>


<div class="figure">
<span style="display:block;" id="fig:pantherFail"></span>
<img src="figures/panther_fail_exp.png" alt="Panther result" width="98%"><p class="caption">
Figure 7.2: <strong>Panther result</strong>
</p>
</div>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="probe-id-conversion-1.html"><span class="header-section-number">6</span> Probe ID conversion</a></div>
<div class="next"><a href="ora-kegg-analysis-1.html"><span class="header-section-number">8</span> ORA KEGG analysis</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#ora-go-analysis-1"><span class="header-section-number">7</span> ORA GO analysis</a></li>
<li><a class="nav-link" href="#go-introduction"><span class="header-section-number">7.1</span> GO intruduction</a></li>
<li>
<a class="nav-link" href="#go-basic-usage"><span class="header-section-number">7.2</span> Basic usage</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#go-additional-arguments"><span class="header-section-number">7.2.1</span> Additional arguments</a></li>
<li><a class="nav-link" href="#go-new-features"><span class="header-section-number">7.2.2</span> New features</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#go-group-enrichment"><span class="header-section-number">7.3</span> Group enrichment analysis</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#comparison-with-one-group"><span class="header-section-number">7.3.1</span> Comparison with one group:</a></li>
<li><a class="nav-link" href="#comparison-with-more-than-one-group"><span class="header-section-number">7.3.2</span> Comparison with more than one group</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#go-advanced-usage"><span class="header-section-number">7.4</span> Advanced usage</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#different-types-of-input-and-backgroud-genes"><span class="header-section-number">7.4.1</span> Different types of input and backgroud genes</a></li>
<li><a class="nav-link" href="#go-simplify"><span class="header-section-number">7.4.2</span> Simplify GO result</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#go-tools-comparsion"><span class="header-section-number">7.5</span> Comparison with other tools</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#gengo-vs-enrichgo"><span class="header-section-number">7.5.1</span> genGO vs enrichGO</a></li>
<li><a class="nav-link" href="#gengo-vs-panther-web"><span class="header-section-number">7.5.2</span> genGO vs Panther web</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Genekitr: Gene Analysis Toolkit based on R</strong>" was written by Yunze Liu. It was last built on 2022-06-27.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
