<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 6 Probe ID conversion | Genekitr: Gene Analysis Toolkit based on R</title>
<meta name="author" content="Yunze Liu">
<meta name="description" content="Probes are used to measure the expression level of many genes. After getting micro array data, the first important thing before downstream analysis maybe probe id conversion. Ensembl database...">
<meta name="generator" content="bookdown 0.26 with bs4_book()">
<meta property="og:title" content="Chapter 6 Probe ID conversion | Genekitr: Gene Analysis Toolkit based on R">
<meta property="og:type" content="book">
<meta property="og:description" content="Probes are used to measure the expression level of many genes. After getting micro array data, the first important thing before downstream analysis maybe probe id conversion. Ensembl database...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 6 Probe ID conversion | Genekitr: Gene Analysis Toolkit based on R">
<meta name="twitter:description" content="Probes are used to measure the expression level of many genes. After getting micro array data, the first important thing before downstream analysis maybe probe id conversion. Ensembl database...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.22/datatables.js"></script><link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Genekitr: Gene Analysis Toolkit based on R</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">🧬 Welcome to genekitr!</a></li>
<li><a class="" href="quick-start.html"><span class="header-section-number">1</span> Quick start</a></li>
<li class="book-part">Part I: Search Information</li>
<li><a class="" href="search-gene-metadata-1.html"><span class="header-section-number">2</span> Search gene metadata</a></li>
<li><a class="" href="search-pubmed-1.html"><span class="header-section-number">3</span> Search PubMed records</a></li>
<li class="book-part">Part II: ID Conversion</li>
<li><a class="" href="gene-id-conversion-1.html"><span class="header-section-number">4</span> Gene ID conversion</a></li>
<li><a class="" href="protein-id-conversion-1.html"><span class="header-section-number">5</span> Protein ID conversion</a></li>
<li><a class="active" href="probe-id-conversion-1.html"><span class="header-section-number">6</span> Probe ID conversion</a></li>
<li class="book-part">Part III: Enrichment Analysis</li>
<li><a class="" href="ora-go-analysis-1.html"><span class="header-section-number">7</span> ORA GO analysis</a></li>
<li><a class="" href="ora-kegg-analysis-1.html"><span class="header-section-number">8</span> ORA KEGG analysis</a></li>
<li><a class="" href="functional-class-scoring-1.html"><span class="header-section-number">9</span> Functional Class Scoring</a></li>
<li class="book-part">Part IV: Plotting</li>
<li><a class="" href="plot-go-kegg-1.html"><span class="header-section-number">10</span> Plot GO and KEGG</a></li>
<li><a class="" href="plot-gsea-1.html"><span class="header-section-number">11</span> Plot GSEA</a></li>
<li><a class="" href="plot-others-1.html"><span class="header-section-number">12</span> Plot others</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="probe-id-conversion-1" class="section level1">
<h1>
<span class="header-section-number">6</span> Probe ID conversion<a class="anchor" aria-label="anchor" href="#probe-id-conversion-1"><i class="fas fa-link"></i></a>
</h1>
<p>Probes are used to measure the expression level of many genes. After getting micro array data, the first important thing before downstream analysis maybe probe id conversion.</p>
<p>Ensembl database regularly maps probes sets against the latest set of transcript models using their own <a href="(https://www.ensembl.org/info/genome/microarray_probe_set_mapping.html)">pipeline</a>.</p>
<p>Here, <code>transProbe</code> utilizes Ensembl up-to-date annotation datasets and could join together with other genekitr functions fluently.</p>
<blockquote>
<p>Notice: now we only support human data</p>
</blockquote>
<div id="transProbe-supported-platforms" class="section level2">
<h2>
<span class="header-section-number">6.1</span> Supported platforms<a class="anchor" aria-label="anchor" href="#transProbe-supported-platforms"><i class="fas fa-link"></i></a>
</h2>
<p>36 probe platforms include Affymetrix, Agilent, Illumina, Codelink and Phalanx.</p>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="va">hsapiens_probe_platform</span>, options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>pageLength <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-cf06e5984f4368f0ba4d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-cf06e5984f4368f0ba4d">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36"],["affy_hc_g110","affy_hg_focus","affy_hg_u133a","affy_hg_u133a_2","affy_hg_u133b","affy_hg_u133_plus_2","affy_hg_u95a","affy_hg_u95av2","affy_hg_u95b","affy_hg_u95c","affy_hg_u95d","affy_hg_u95e","affy_hta_2_0","affy_ht_hg_u133_plus_pm","affy_hugenefl","affy_hugene_1_0_st_v1","affy_hugene_2_0_st_v1","affy_hugene_2_1_st_v1","affy_primeview","affy_u133_x3p","illumina_humanht_12_v3","illumina_humanht_12_v4","illumina_humanref_8_v3","illumina_humanwg_6_v1","illumina_humanwg_6_v2","illumina_humanwg_6_v3","agilent_cgh_44b","agilent_gpl26966","agilent_gpl6848","agilent_sureprint_g3_ge_8x60k","agilent_sureprint_g3_ge_8x60k_v2","agilent_wholegenome","agilent_wholegenome_4x44k_v1","agilent_wholegenome_4x44k_v2","codelink_codelink","phalanx_onearray"],["Affymetrix HC_G110 Array","Affymetrix HG-Focus Array","Affymetrix HG-U133A Array","Affymetrix HG-U133A_2 Array","Affymetrix HG-U133B Array","Affymetrix Human Genome U133 Plus 2.0 Array","Affymetrix HG_U95A Array","Affymetrix Human Genome U95 Set","Affymetrix HG_U95B Array","Affymetrix HG_U95C Array","Affymetrix HG_U95D Array","Affymetrix HG_U95E Array","Affymetrix HTA-20 Array","Affymetrix HT_HG-U133_Plus_PM Array","Affymetrix HuGeneFL Genome Array","Affymetrix HuGene-10-st-v1 Array","Affymetrix HuGene-20-st-v1 Array","Affymetrix HuGene-21-st-v1 Array","Human Genome PrimeView Array","Affymetrix Human X3P Array","Illumina HumanHT12v3","Illumina HumanHT12v4","Illumina HumanRef-8 v3.0 expression beadchip","Illumina HumanWG6v1","Illumina HumanWG6v2","Illumina HumanWG6v3","Agilent CGH 44b probe","Agilent GPL26966","Agilent GPL6848","Agilent Sureprint G3 GE 8x60k","Agilent SurePrint G3 GE 8x60k v2","Agilent WholeGenome","Agilent WholeGenome 4x44k v1","Agilent WholeGenome 4x44k v2","Codelink microarray","Phalanx OneArray hybridization kit"],["hcg110.db","hgfocus.db","hgu133a.db","hgu133a2.db","hgu133b.db","hgu133plus2.db","hgu95a.db","hgu95av2.db","hgu95b.db","hgu95c.db","hgu95d.db","hgu95e.db","hta20probeset.db","hthgu133pluspm.db",null,"hugene10sttranscriptcluster.db","hugene20sttranscriptcluster.db","hugene21sttranscriptcluster.db",null,"u133x3p.db","illuminaHumanv3.db","illuminaHumanv4.db",null,"illuminaHumanv1.db","illuminaHumanv2.db",null,null,null,null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Platform<\/th>\n      <th>Description<\/th>\n      <th>Bioc_anno<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="probe-basic-usage" class="section level2">
<h2>
<span class="header-section-number">6.2</span> Basic usage<a class="anchor" aria-label="anchor" href="#probe-basic-usage"><i class="fas fa-link"></i></a>
</h2>
<p><code>transProbe</code> has four arguments:</p>
<ul>
<li>
<code>id</code>: probe id</li>
<li>
<code>transTo</code>: the conversion target type. User could select <strong>more than one</strong> from “symbol”, “entrez”, “ensembl” or “uniprot”.</li>
<li>
<code>org</code>: human (default)</li>
<li>
<code>platform</code>: Probe platform. If not given, program will detect automatically.</li>
</ul>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">deg</span>, package <span class="op">=</span> <span class="st">"genekitr"</span><span class="op">)</span>
<span class="va">probes</span> <span class="op">&lt;-</span> <span class="va">deg</span><span class="op">$</span><span class="va">probe_id</span>

<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/transProbe.html">transProbe</a></span><span class="op">(</span><span class="va">probes</span>, transTo <span class="op">=</span> <span class="st">"ensembl"</span>, org <span class="op">=</span> <span class="st">"human"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">probes</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "8144866" "8066431" "8022674" "7925531" "8081358" "7979931"</code></pre>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 2
## # Groups:   probe_id [6]
##   probe_id ensembl        
##   &lt;chr&gt;    &lt;chr&gt;          
## 1 8144866  ENSG00000156006
## 2 8066431  ENSG00000196839
## 3 8022674  ENSG00000170558
## 4 7925531  ENSG00000275199
## 5 8081358  ENSG00000256628
## 6 7979931  ENSG00000133997</code></pre>
</div>
<div id="convert-probe-types" class="section level2">
<h2>
<span class="header-section-number">6.3</span> Convert to more than one type<a class="anchor" aria-label="anchor" href="#convert-probe-types"><i class="fas fa-link"></i></a>
</h2>
<p>User could define <code>transTo</code> as many types:</p>
<div class="sourceCode" id="cb122"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/genekitr/man/transProbe.html">transProbe</a></span><span class="op">(</span><span class="va">probes</span>, transTo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ensembl"</span>,<span class="st">"symbol"</span><span class="op">)</span>, org <span class="op">=</span> <span class="st">"human"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">res2</span><span class="op">)</span></code></pre></div>
<pre><code>##       probe_id         ensembl     symbol
## 9023   8144866 ENSG00000156006       NAT2
## 15007  8066431 ENSG00000196839        ADA
## 11595  8022674 ENSG00000170558       CDH2
## 17512  7925531 ENSG00000275199       AKT3
## 17126  8081358 ENSG00000256628 ZBTB11-AS1
## 6218   7979931 ENSG00000133997       MED6</code></pre>
<p>Let’s take the first record as <a href="example:%5BENSG00000228794%5D(https://www.ensembl.org/id/ENSG00000228794" class="uri">example:[ENSG00000228794](https://www.ensembl.org/id/ENSG00000228794</a>) has 38 transcripts.</p>
<p>When we look at its second transcript <a href="http://asia.ensembl.org/Homo_sapiens/Transcript/Oligos?db=core;g=ENSG00000228794;r=1:825138-859446;t=ENST00000445118">ENST00000445118.7</a>, probe <code>7896759</code> is listed in <code>AFFY HuGene-1_0-st-v1</code> database.</p>

</div>
</div>



  <div class="chapter-nav">
<div class="prev"><a href="protein-id-conversion-1.html"><span class="header-section-number">5</span> Protein ID conversion</a></div>
<div class="next"><a href="ora-go-analysis-1.html"><span class="header-section-number">7</span> ORA GO analysis</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#probe-id-conversion-1"><span class="header-section-number">6</span> Probe ID conversion</a></li>
<li><a class="nav-link" href="#transProbe-supported-platforms"><span class="header-section-number">6.1</span> Supported platforms</a></li>
<li><a class="nav-link" href="#probe-basic-usage"><span class="header-section-number">6.2</span> Basic usage</a></li>
<li><a class="nav-link" href="#convert-probe-types"><span class="header-section-number">6.3</span> Convert to more than one type</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Genekitr: Gene Analysis Toolkit based on R</strong>" was written by Yunze Liu. It was last built on 2022-06-23.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
